{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "f7245f5c-6215-45c8-a969-0f8eac008ea1",
   "metadata": {},
   "outputs": [],
   "source": [
    "#data\n",
    "import pandas as pd\n",
    "import os\n",
    "import statsmodels.stats.multitest as multi\n",
    "import numpy as np\n",
    "import json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "b45da84d-f6d8-4f4d-8924-7a7334a9cf69",
   "metadata": {},
   "outputs": [],
   "source": [
    "#plotting\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import matplotlib\n",
    "matplotlib.rcParams['pdf.fonttype'] = 42\n",
    "matplotlib.rcParams['ps.fonttype'] = 42"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "3c75668e-6c3f-438d-8c27-764e8cc310a6",
   "metadata": {},
   "outputs": [],
   "source": [
    "#statistics\n",
    "import lifelines\n",
    "from lifelines import CoxPHFitter\n",
    "from lifelines import KaplanMeierFitter\n",
    "import statsmodels.stats.multitest as multi\n",
    "from sklearn.linear_model import LogisticRegression\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "from lifelines.statistics import logrank_test\n",
    "from scipy.stats import mannwhitneyu\n",
    "import statsmodels.formula.api as smf\n",
    "from scipy import stats\n",
    "from sklearn.model_selection import train_test_split\n",
    "from sklearn.linear_model import Lasso\n",
    "import sklearn.metrics as metrics\n",
    "import xgboost\n",
    "from sklearn.model_selection import cross_val_score\n",
    "from sklearn.model_selection import RepeatedKFold"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "9fd15a7f-6f46-4dec-bbcc-3a734890aabe",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('/cellar/users/mpagadal/resources/tcga/ensembl_map.json', 'r') as f:\n",
    "    ensembl = json.load(f)\n",
    "ensembl={k.split(\".\")[0]:v for k,v in ensembl.items()}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "1db56e49-2d3b-4bc5-b3f2-d98827aa84ce",
   "metadata": {},
   "outputs": [],
   "source": [
    "random_state=75"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "e313721e-9035-4047-8d89-cfff879360d5",
   "metadata": {},
   "outputs": [],
   "source": [
    "def make_surv_df(surv,surv_type,raw):\n",
    "    \n",
    "    '''\n",
    "    make TCGA survival dataframe\n",
    "    '''\n",
    "    \n",
    "    #get survival dataframe\n",
    "    surv=pd.read_csv(surv,index_col=0)\n",
    "    surv=surv.rename(columns={\"bcr_patient_barcode\":\"FID\"})\n",
    "    surv=surv[surv[\"Redaction\"].isnull()]\n",
    "    \n",
    "    #get phenos\n",
    "    raw=pd.read_csv(raw,delim_whitespace=True,index_col=0)\n",
    "    raw=raw.iloc[:,5:]\n",
    "    raw=raw.reset_index()\n",
    "    raw=raw.rename(columns={\"index\":\"FID\"})\n",
    "\n",
    "    #combine dataframes\n",
    "    full_surv=pd.merge(surv[[\"FID\",surv_type,surv_type+\".time\",\"age_at_initial_pathologic_diagnosis\",\"race\",\"gender\",\"ajcc_pathologic_tumor_stage\"]], raw, on=\"FID\", how=\"left\")\n",
    "    #full_surv=full_surv[full_surv[surv_type+\".time\"]<1825] #censor at 5 years\n",
    "    #get patients with Stage designations\n",
    "    full_surv=full_surv[full_surv[\"ajcc_pathologic_tumor_stage\"].str.contains(\"Stage\")]\n",
    "    full_surv=full_surv[~(full_surv[\"ajcc_pathologic_tumor_stage\"].isin([\"Stage X\"]))]\n",
    "    #split A,B,C from stage designations\n",
    "    full_surv[\"ajcc_pathologic_tumor_stage\"]=full_surv[\"ajcc_pathologic_tumor_stage\"].str.split(\"A\").str[0]\n",
    "    full_surv[\"ajcc_pathologic_tumor_stage\"]=full_surv[\"ajcc_pathologic_tumor_stage\"].str.split(\"B\").str[0]\n",
    "    full_surv[\"ajcc_pathologic_tumor_stage\"]=full_surv[\"ajcc_pathologic_tumor_stage\"].str.split(\"C\").str[0]\n",
    "    print(full_surv.shape)\n",
    "    full_surv=full_surv.join(pd.get_dummies(full_surv[\"ajcc_pathologic_tumor_stage\"]))\n",
    "    #remap gender\n",
    "    full_surv[\"gender\"]=full_surv[\"gender\"].map({\"MALE\":0,\"FEMALE\":1})\n",
    "    \n",
    "    full_surv=full_surv[~full_surv[surv_type].isnull()]\n",
    "    full_surv=full_surv[~full_surv[surv_type+\".time\"].isnull()]\n",
    "    \n",
    "    full_surv=full_surv[~full_surv[\"age_at_initial_pathologic_diagnosis\"].isnull()]\n",
    "    \n",
    "    #map to cancer type\n",
    "    canc = pd.read_csv(\"/cellar/users/mpagadal/gtype.meta\", delimiter=\"\\t\", header=None)\n",
    "    mp = canc.set_index(2)[0].to_dict()\n",
    "    full_surv[\"cancer\"]=full_surv[\"FID\"].map(mp)\n",
    "    full_surv=full_surv[~full_surv[\"cancer\"].isnull()]\n",
    "    \n",
    "    full_surv['ajcc_pathologic_tumor_stage'] = full_surv['ajcc_pathologic_tumor_stage'].replace({'Stage 0':0,'Stage I': 1,'Stage II': 2,'Stage III': 3,'Stage IV': 4})\n",
    "    \n",
    "    return(full_surv)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "dbd23280-3c78-4bbf-85ad-3c7df8f46c53",
   "metadata": {},
   "outputs": [],
   "source": [
    "def run_surv_plot(df_annot,surv_df,surv_type,out):\n",
    "    \n",
    "    '''\n",
    "    plot kaplan meier for each snp in df_annot\n",
    "    '''\n",
    "    for i,row in df_annot.iterrows():\n",
    "        try:\n",
    "            surv_df_canc=surv_df[surv_df[\"cancer\"]==row[\"cancer\"]]\n",
    "            x=[x for x in surv_df.columns if row[\"snps\"].rsplit(\":\",2)[0] in x][0]\n",
    "            rs=x.rsplit(\":\",2)[0]\n",
    "        \n",
    "            allele1=x.split(\":\")[2]\n",
    "            allele2=str(x.split(\"_\")[0]).split(\":\")[3]\n",
    "            minor=x.split(\"_\")[1]\n",
    "            if allele1 == minor:\n",
    "                major=allele2\n",
    "            else:\n",
    "                major=allele1\n",
    "            \n",
    "            fig=plt.figure(figsize=(12,5))\n",
    "            ax1 = plt.subplot(1,2,1)\n",
    "            \n",
    "            groups = surv_df_canc[x]\n",
    "            ix0 = (groups == 0)\n",
    "            ix1 = (groups == 1)\n",
    "            ix2 = (groups == 2)\n",
    "\n",
    "            kmf = KaplanMeierFitter()\n",
    "            kmf.fit(surv_df_canc[surv_type+'.time'][ix2], surv_df_canc[surv_type][ix2],label=minor+minor)\n",
    "            kmf.plot(color='green', ci_show=False, ax=ax1)\n",
    "            \n",
    "            kmf.fit(surv_df_canc[surv_type+'.time'][ix1], surv_df_canc[surv_type][ix1],label=major+minor)\n",
    "            kmf.plot(ax=ax1, color='orange', ci_show=False)\n",
    "    \n",
    "            kmf.fit(surv_df_canc[surv_type+'.time'][ix0], surv_df_canc[surv_type][ix0],label=major+major)\n",
    "            kmf.plot(ax=ax1, color='blue', ci_show=False)\n",
    "\n",
    "            results = logrank_test(surv_df_canc[surv_type+'.time'][ix0], surv_df_canc[surv_type+'.time'][ix2],event_observed_A=surv_df_canc[surv_type][ix0], event_observed_B=surv_df_canc[surv_type][ix2], alpha=.95) \n",
    "            plt.title('{}\\n{}\\nLog-rank test: p<{:.3}'.format(x,row[\"cancer\"],results.p_value))\n",
    "\n",
    "            plt.ylabel('% survival')\n",
    "            plt.xlabel(surv_type+' time (days)')\n",
    "            plt.legend(frameon=False)\n",
    "\n",
    "            plt.tight_layout()\n",
    "\n",
    "            plt.savefig(out+surv_type+\".\"+x+\".\"+row[\"cancer\"]+'.pdf')\n",
    "            plt.close()\n",
    "        except:\n",
    "            pass\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "e2041278-00e1-40b9-a0cc-ebbf2fda182e",
   "metadata": {},
   "outputs": [],
   "source": [
    "def fdr(df,pseudo,p):\n",
    "    \n",
    "    '''\n",
    "    calculate Benjamini-Hochberg FDR\n",
    "    pseudo: True or False\n",
    "    '''\n",
    "    \n",
    "    #add pseudocounta\n",
    "    if pseudo == True:\n",
    "        df[p]=df[p]+0.000000000001\n",
    "    \n",
    "    #calculate benjamin-hochberg fdr\n",
    "    probabilities = df[p].to_numpy()\n",
    "    report = multi.multipletests(probabilities,\n",
    "                alpha=0.05,\n",
    "                method=\"fdr_bh\",\n",
    "                is_sorted=False,\n",
    "            )\n",
    "    discoveries = report[1]\n",
    "    df[\"fdr\"]=discoveries\n",
    "    df[\"fdr\"]=pd.to_numeric(df[\"fdr\"])\n",
    "    df[\"-log10(fdr)\"]=-np.log10(df[\"fdr\"])\n",
    "    \n",
    "    return(df)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7ab483fa-858f-493d-8ee0-0d828d62682c",
   "metadata": {},
   "source": [
    "### Get associations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "08d12f2f-f1b5-4e6c-a90a-12cb965d7920",
   "metadata": {},
   "outputs": [],
   "source": [
    "snps=pd.read_csv(\"../tables/time-germline.csv\",delimiter=\"\\t\")\n",
    "snps[\"snp_noallele\"]=snps[\"proxy_variant\"].str.rsplit(\":\",2).str[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "a2d735c3-069d-4c9c-b9be-6c8a97fdd3f5",
   "metadata": {},
   "outputs": [],
   "source": [
    "fdr_thresh=0.05"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "8688bcdc-0ec5-4e78-939a-45e0af59f5e4",
   "metadata": {},
   "outputs": [],
   "source": [
    "surv_fdr=pd.read_csv(\"../tables/supplementary_table_11.csv\",sep=\",\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "697739b3-ae0d-40ee-a308-a1e23939946e",
   "metadata": {},
   "outputs": [],
   "source": [
    "prog_snps=surv_fdr[surv_fdr[\"logrank p\"]<fdr_thresh][\"snps\"].unique().tolist()\n",
    "prog_snps=[x.split(\"_\")[0] for x in prog_snps]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "5378f414-6720-4873-b916-88e5f9c2ade1",
   "metadata": {},
   "outputs": [],
   "source": [
    "#check how many snps are associated with survival in multiple cancer types\n",
    "surv_number=surv_fdr[surv_fdr[\"logrank p\"]<fdr_thresh][[\"snps\",\"cancer\"]].drop_duplicates().groupby([\"snps\"]).size().reset_index()\n",
    "surv_number=surv_number.sort_values(by=0,ascending=False)\n",
    "surv_number[\"snp_noallele\"]=surv_number[\"snps\"].str.rsplit(\":\",2).str[0]\n",
    "snp_sig=snps[snps[\"snp_noallele\"].isin(surv_number[surv_number[0]>1][\"snp_noallele\"].tolist())]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "47168e05-e2d0-4136-8836-324b95fc2b06",
   "metadata": {},
   "outputs": [],
   "source": [
    "snp_sig=snp_sig.loc[~snp_sig[\"proxy_variant\"].duplicated()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "9c4dce81-0078-48d3-8388-6c0157557dc0",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3146: DtypeWarning: Columns (8,9,27,42,44,45,46,47,48,49,50,51,52,53,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,161,162,163,164,165,166,167,168,169,170,171,172,173,174,175,176,177,178,179,180,181,182,183,184,185,186,187,188,189,190,191,192,193,194,195,196,197,198,199,200,201,202,203,204,205,206,207,208,209,210,211,212,213,214,215,216,217,218,219,220,221,222,223,224,225,226,227,228,229,230,231,232,233,234,235,236,237,238,239,240,241,242,243,244,245,246,247,248,249,250,251,252,253,254,255,256,257,258,259,260,261,262,263,264,265,266,267,268,269,270,271,272,273,274,275,276,277,278,279,280,281,282,283,284,285,286,287,288,289,290,291,292,293,294,295,296,297,298,299,300,301,302,303,304,305,306,307,308,309,310,311,312,313,314,315,316,317,318,319,320,321,322,323,324,325,326,327,328,329,330,333,334,335,336,338,339,340,341,342,343,344,345,346,347,348,349,350,351,352,353,354,355,356,357,358,359,360,361,362,363,364,365,366,367,368,369,370,371,372,373,374,375,376,377,378,379,380,381,382,383,384,385,386,387,388,389,390,391,392,393,394,395,396,397,398,399,400,401,402,403,404,405,406,407,408,409,410,411,412,413,414,415,416,417,418,419,420,421,422,423,424,425,426,427,428,429,430,431,432,433,434,435,436,437,438,439,440,441,442,443,444,445,446,447,448,449,450,451,452,453,454,455,456,457,458,459,460,461,462,463,464,465,466,467,468,469,470,471,472,473,474,475,476,477,478,479,480,481,482,483,484,485,486,487,488,489,490,491,492,493,494,495,496,497,498,499,500,501,502,503,504,505,506,507,508,509,510,511,512,513,514,515,516,517,518,519,520,521,522,523,524,525,526,527,528,529,530,531,532,533,534,535,536,537,538,539,540,541,542,543,544,545,546,547,548,549,550,551,552,553,554,555,556,557,558,559,560,561,562,563,564,565,566,567,568,569,570,571,572,573,574,575,576,577,578,579,580,581,582,583,584,585,586,587,588,589,590,591,592,593,594,595,596,597,598,599,600,601,602,603,604,605,606,607,608,609,610,611,612,613,614,615,616,617,618,619,620,621,622,623,624,625,626,627,628,629,630,631,632,633,634,635,636,637,639,640,641,642,643,644,645,646,647,648,649,650,651,652,653,654,655,656,657,658,659,660,661,662,663,664,665,666,667,668,669,670,671,672,673,674,675,676,677,678,679,680,681,682,683,684,685,686,687,688,689,690,691,692,693,694,695,696,697,698,699,700,701,702,703,704,705,706,707,708,709,710,711,712,713,714,715,716,717,718,719,720,721,722,723,724,725,726,727,728,729,730,731,732,733,734,735,736,737,738,739,740,741,742,743,744) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "history_of_neoadjuvant_treatment\n",
      "radiation_therapy\n",
      "pathologic_T\n",
      "pathologic_M\n",
      "pathologic_N\n",
      "pathologic_stage\n",
      "lymph_node_examined_count\n",
      "number_of_lymphnodes_positive_by_he\n",
      "neoplasm_histologic_grade\n",
      "height\n",
      "weight\n",
      "tobacco_smoking_history\n",
      "targeted_molecular_therapy\n",
      "menopause_status\n",
      "breast_carcinoma_progesterone_receptor_status\n",
      "breast_carcinoma_estrogen_receptor_status\n",
      "lab_proc_her2_neu_immunohistochemistry_receptor_status\n",
      "lab_procedure_her2_neu_in_situ_hybrid_outcome_type\n",
      "diagnosis\n",
      "alcohol_history_documented\n",
      "synchronous_colon_cancer_present\n",
      "hypertension\n",
      "seizure_history\n",
      "history_ionizing_rt_to_head\n",
      "headache_history\n",
      "extrathyroid_carcinoma_present_extension_status\n",
      "gleason_score\n",
      "diagnosis_subtype\n",
      "history_hepato_carcinoma_risk_factor\n",
      "ct_scan\n",
      "history_of_undescended_testis\n",
      "history_hypospadias\n",
      "history_myasthenia_gravis\n",
      "mitotane_therapy\n",
      "ebv_antibody_status\n",
      "relative_family_cancer_history\n",
      "hist_hepato_carc_fact\n",
      "creatinine_value_in_mg_dl\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/ipykernel_launcher.py:21: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/ipykernel_launcher.py:22: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/ipykernel_launcher.py:23: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/ipykernel_launcher.py:24: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/pandas/core/indexing.py:1765: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/ipykernel_launcher.py:52: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/ipykernel_launcher.py:55: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/ipykernel_launcher.py:56: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/ipykernel_launcher.py:57: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/ipykernel_launcher.py:58: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/ipykernel_launcher.py:59: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/ipykernel_launcher.py:60: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/ipykernel_launcher.py:61: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/ipykernel_launcher.py:62: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "pathologic_T\n",
      "pathologic_M\n",
      "pathologic_N\n",
      "neoplasm_histologic_grade\n",
      "tobacco_smoking_history\n",
      "menopause_status\n",
      "extrathyroid_carcinoma_present_extension_status\n",
      "diagnosis_subtype\n",
      "history_hepato_carcinoma_risk_factor\n",
      "hist_hepato_carc_fact\n"
     ]
    }
   ],
   "source": [
    "surv=pd.read_csv(\"../data/survival/clinical_PANCAN_patient_with_followup.tsv\",delimiter=\"\\t\",encoding=\"unicode_escape\")\n",
    "keep=pd.read_csv(\"../data/survival/surv_var_keep_MP.csv\") #keep only curated survival variables\n",
    "surv_clean=surv[[\"acronym\"]+keep[\"var\"].tolist()]\n",
    "\n",
    "total_counts=pd.DataFrame()\n",
    "for x in surv_clean.columns[2:]:\n",
    "    surv_not_null=surv_clean[~surv_clean[x].isnull()]\n",
    "    print(x)\n",
    "    try:\n",
    "        surv_not_null=surv_not_null[~(surv_not_null[x].str.contains(\"Not Av\"))]\n",
    "        surv_not_null=surv_not_null[~(surv_not_null[x].str.contains(\"Not Ap\"))]\n",
    "        surv_not_null=surv_not_null[~(surv_not_null[x].str.contains(\"Unk\"))]\n",
    "        df_counts=surv_not_null[\"acronym\"].value_counts().reset_index()\n",
    "        df_counts[\"var\"]=x\n",
    "        total_counts=total_counts.append(df_counts)\n",
    "    except:\n",
    "        df_counts=surv_not_null[\"acronym\"].value_counts().reset_index()\n",
    "        df_counts[\"var\"]=x\n",
    "        total_counts=total_counts.append(df_counts)\n",
    "supp_table=total_counts.groupby('index')['var'].apply(lambda x: ','.join(x)).reset_index()\n",
    "supp_table.loc[14][\"var\"]='history_of_neoadjuvant_treatment,radiation_therapy,height,weight,T2_pathologic_T,T3_pathologic_T,T4_pathologic_T,M1_pathologic_M,N1_pathologic_N,G1_neoplasm_histologic_grade,G2_neoplasm_histologic_grade,G3_neoplasm_histologic_grade,G4_neoplasm_histologic_grade,alcohol,NAFLD,hepatitis'\n",
    "supp_table.loc[15][\"var\"]='radiation_therapy,targeted_molecular_therapy,T2_pathologic_T,T3_pathologic_T,T4_pathologic_T,M1_pathologic_M,N1_pathologic_N,N2_pathologic_N,N3_pathologic_N,Current smoker_tobacco_smoking_history,Lifelong Non-smoker_tobacco_smoking_history'\n",
    "supp_table.loc[26][\"var\"]='radiation_therapy,history_of_undescended_testis,T2_pathologic_T,M1_pathologic_M,N1_pathologic_N,N2_pathologic_N'\n",
    "supp_table.loc[29][\"var\"]='radiation_therapy,height,weight,hypertension,G2_neoplasm_histologic_grade,G3_neoplasm_histologic_grade,High Grade_neoplasm_histologic_grade,Peri (6-12 months since last menstrual period)_menopause_status,Post (prior bilateral ovariectomy OR >12 mo since LMP with no prior hysterectomy)_menopause_status,Pre (<6 months since LMP AND no prior bilateral ovariectomy AND not on estrogen replacement)_menopause_status'\n",
    "\n",
    "for x in surv_clean.columns:\n",
    "    try:\n",
    "        m = surv_clean[x].str.contains('\\[', na=False)\n",
    "        surv_clean.loc[m,x] = np.nan\n",
    "    except:\n",
    "        pass\n",
    "\n",
    "# map variables to category\n",
    "mp_map={'Equivocal':np.nan,\n",
    " 'Indeterminate':np.nan,\n",
    " 'NO':0,\n",
    " 'Negative':0,\n",
    " 'No':0,\n",
    " 'Not Performed':np.nan,\n",
    " 'Positive':1,\n",
    " 'YES':1,\n",
    " 'Yes':1,\n",
    " 'Yes, Pharmaceutical Treatment Prior to Resection':1,\n",
    " 'Yes, Radiation Prior to Resection':1,\n",
    " 'Yes; bilateral':1,\n",
    " 'Yes; left testicle only':1,\n",
    " 'Yes; right testicle only':1,\n",
    " 'negative':0,\n",
    " 'positive':1,\n",
    "}\n",
    "for var in keep[keep[\"coding\"]==\"map\"][\"var\"].unique():\n",
    "    surv_clean[var]=surv_clean[var].map(mp_map)  \n",
    "\n",
    "#clean staging\n",
    "surv_clean[\"pathologic_T\"]=surv_clean[\"pathologic_T\"].str[0:2]\n",
    "surv_clean[\"pathologic_N\"]=surv_clean[\"pathologic_N\"].str[0:2]\n",
    "surv_clean[\"pathologic_M\"]=surv_clean[\"pathologic_M\"].str[0:2]\n",
    "surv_clean[\"pathologic_stage\"]=surv_clean[\"pathologic_stage\"].str.replace(\"A\",\"\")\n",
    "surv_clean[\"pathologic_stage\"]=surv_clean[\"pathologic_stage\"].str.replace(\"B\",\"\")\n",
    "surv_clean[\"pathologic_stage\"]=surv_clean[\"pathologic_stage\"].str.replace(\"C\",\"\")\n",
    "surv_clean[\"pathologic_stage\"]=surv_clean[\"pathologic_stage\"].replace(\"Stage X\",np.nan)\n",
    "surv_clean['pathologic_stage'] = surv_clean['pathologic_stage'].replace({'Stage 0':0,'Stage I': 1,'Stage II': 2,'Stage III': 3,'Stage IV': 4,'Stage IS':0,'I or II NOS':0})\n",
    "\n",
    "#dummy code specified variables\n",
    "mp_dummy={}\n",
    "for x in keep[keep[\"coding\"]==\"dummy\"][\"var\"].unique():\n",
    "    print(x)\n",
    "    df_dummy=pd.get_dummies(surv_clean[x])\n",
    "    df_dummy.columns=[col+\"_\"+x for col in df_dummy.columns]\n",
    "    mp_dummy[x]=df_dummy.columns.tolist()\n",
    "    surv_clean=surv_clean.join(df_dummy)\n",
    "    del surv_clean[x]\n",
    "    if \"None\" in surv_clean.columns:\n",
    "        del surv_clean[\"None\"]\n",
    "    if \"Other\" in surv_clean.columns:\n",
    "        del surv_clean[\"Other\"]\n",
    "    \n",
    "surv=pd.read_csv(\"/cellar/users/mpagadal/resources/from-labmembers/andrea/Liu2018.TCGA_survival.csv\")\n",
    "surv=surv[[\"bcr_patient_barcode\",\"gender\",\"age_at_initial_pathologic_diagnosis\",\"type\",\"OS\",\"OS.time\",\"PFI\",\"PFI.time\"]]\n",
    "surv=pd.merge(surv,surv_clean,on=\"bcr_patient_barcode\",how=\"left\")\n",
    "surv[\"gender\"]=surv[\"gender\"].map({\"MALE\":0,\"FEMALE\":1})\n",
    "\n",
    "del surv[\"T0_pathologic_T\"]\n",
    "del surv[\"TX_pathologic_T\"]\n",
    "del surv[\"Ti_pathologic_T\"]\n",
    "del surv[\"T1_pathologic_T\"]\n",
    "\n",
    "del surv[\"MX_pathologic_M\"]\n",
    "del surv[\"cM_pathologic_M\"]\n",
    "del surv[\"M0_pathologic_M\"]\n",
    "\n",
    "del surv[\"N0_pathologic_N\"]\n",
    "del surv[\"NX_pathologic_N\"]\n",
    "\n",
    "del surv[\"Indeterminate (neither Pre or Postmenopausal)_menopause_status\"]\n",
    "del surv[\"No History of Primary Risk Factors_hist_hepato_carc_fact\"]\n",
    "del surv[\"Other_hist_hepato_carc_fact\"]\n",
    "del surv[\"lymph_node_examined_count\"]\n",
    "del surv[\"synchronous_colon_cancer_present\"]\n",
    "del surv[\"pathologic_stage\"]\n",
    "\n",
    "del surv[\"GX_neoplasm_histologic_grade\"]\n",
    "del surv[\"G1_neoplasm_histologic_grade\"]\n",
    "\n",
    "del surv[\"Current Reformed Smoker, Duration Not Specified_tobacco_smoking_history\"]\n",
    "del surv[\"Current reformed smoker for < or = 15 years_tobacco_smoking_history\"]\n",
    "del surv[\"Current reformed smoker for > 15 years_tobacco_smoking_history\"]\n",
    "del surv[\"history_of_neoadjuvant_treatment\"]\n",
    "\n",
    "surv[\"alcohol\"]=surv[[x for x in surv.columns if \"Alcohol consumption\" in x]].sum(axis=1)\n",
    "surv[\"NAFLD\"]=surv[[x for x in surv.columns if \"Nonalcoholic Fatty Liver Diseas\" in x]].sum(axis=1)\n",
    "surv[\"hepatitis\"]=surv[[x for x in surv.columns if \"Hepatitis\" in x]].sum(axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "801bb32b-b0d9-4171-9d59-b5cb424b20ba",
   "metadata": {},
   "outputs": [],
   "source": [
    "#merge snps onto clinical file\n",
    "raw=pd.read_csv(\"../data/genotypes/all.time.raw\",delim_whitespace=True)\n",
    "raw=raw.rename(columns={\"FID\":\"bcr_patient_barcode\"})\n",
    "surv=pd.merge(surv,raw,on=\"bcr_patient_barcode\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "86a8796a-1992-4da4-a45f-595e3b94c915",
   "metadata": {},
   "outputs": [],
   "source": [
    "#get snps\n",
    "prog_snps=surv_fdr[surv_fdr[\"logrank p\"]<fdr_thresh][\"snps\"].unique().tolist()\n",
    "prog_snps=[x.split(\"_\")[0] for x in prog_snps]\n",
    "\n",
    "rsid_lst=[]\n",
    "for snp in prog_snps:\n",
    "    try:\n",
    "        rsid_lst.append([x for x in surv.columns if snp.rsplit(\":\",2)[0] in x][0])\n",
    "    except:\n",
    "        print(snp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "557642e1-1afc-40e1-880b-a1e3a05d4806",
   "metadata": {},
   "outputs": [],
   "source": [
    "canc_counts=surv_clean[\"acronym\"].value_counts().reset_index()\n",
    "canc_keep=canc_counts[canc_counts[\"acronym\"]>100][\"index\"].tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "fee0a594-49ba-4932-9c44-da932e4f9cae",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(6875, 976)\n",
      "(6875, 976)\n"
     ]
    }
   ],
   "source": [
    "#Create OS surv dataframe and split 70/30 by each cancer type\n",
    "os_surv=make_surv_df(\"/cellar/users/mpagadal/resources/from-labmembers/andrea/Liu2018.TCGA_survival.csv\",\"OS\",\"/cellar/users/mpagadal/projects/germline-immune/data/genotypes/all.time.raw\")\n",
    "pfi_surv=make_surv_df(\"/cellar/users/mpagadal/resources/from-labmembers/andrea/Liu2018.TCGA_survival.csv\",\"PFI\",\"/cellar/users/mpagadal/projects/germline-immune/data/genotypes/all.time.raw\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "7ddecb96-dc6f-4e21-b986-569308cd15be",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['ACC', 'BLCA', 'BRCA', 'CHOL', 'COAD', 'ESCA', 'HNSC', 'KICH',\n",
       "       'KIRC', 'KIRP', 'LIHC', 'LUAD', 'LUSC', 'MESO', 'PAAD', 'READ',\n",
       "       'SKCM', 'STAD', 'TGCT', 'THCA', 'UVM'], dtype=object)"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "np.unique(os_surv[\"cancer\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ade8f043-0d7c-4002-ad6a-8de02bd0f177",
   "metadata": {
    "tags": []
   },
   "source": [
    "### PFI survival analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "c8bd1f0c-a7c9-4a0c-871b-b315a9262abd",
   "metadata": {},
   "outputs": [],
   "source": [
    "pfi_test=pd.DataFrame()\n",
    "pfi_train=pd.DataFrame()\n",
    "\n",
    "for cancer in np.unique(pfi_surv[\"cancer\"]):\n",
    "    \n",
    "    temp_train,temp_test=train_test_split(pfi_surv[pfi_surv[\"cancer\"]==cancer],test_size=0.3,random_state=random_state)\n",
    "    pfi_test=pfi_test.append(temp_test)\n",
    "    pfi_train=pfi_train.append(temp_train)\n",
    "\n",
    "surv_filt=surv.drop([\"gender\",\"age_at_initial_pathologic_diagnosis\",\"type\",\"OS\",\"OS.time\",\"PFI\",\"PFI.time\",\"acronym\"],axis=1)\n",
    "surv_filt=surv_filt.drop(surv_filt.filter(like=\":\").columns.values,axis=1)\n",
    "\n",
    "#merge additional covariates into df\n",
    "pfi_test=pd.merge(pfi_test,surv_filt,left_on=\"FID\",right_on=\"bcr_patient_barcode\")\n",
    "pfi_train=pd.merge(pfi_train,surv_filt,left_on=\"FID\",right_on=\"bcr_patient_barcode\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "07ff0a04-4880-4167-b1c5-2f4b6fdd1e4e",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BLCA\n",
      "BRCA\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/miniconda3/envs/ML/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CESC\n",
      "COAD\n",
      "ESCA\n",
      "GBM\n",
      "HNSC\n"
     ]
    }
   ],
   "source": [
    "#run coxph analysis on test datasets, keep cov column for analysis on training datasets\n",
    "\n",
    "df_coxph = pd.DataFrame()\n",
    "cancer_test=[]\n",
    "cov_cols=[]\n",
    "\n",
    "for i,row in supp_table[supp_table[\"index\"].isin(canc_keep)].iterrows():\n",
    "    print(row[\"index\"])\n",
    "    \n",
    "    for snp in rsid_lst:\n",
    "        temp_df = pfi_test[pfi_test[\"cancer\"]==row[\"index\"]]\n",
    "        temp_df=temp_df[~temp_df[snp].isnull()]\n",
    "        \n",
    "        if len(temp_df[snp].value_counts())>1:       \n",
    "            cov_col=row[\"var\"].split(\",\")\n",
    "            for x in cov_col:\n",
    "                if x in [x for x in mp_dummy.keys()]:\n",
    "                    cov_col=cov_col+mp_dummy[x]\n",
    "            cov_col=[x for x in cov_col if x in surv.columns]\n",
    "            cov_col=[x for x in cov_col if len(temp_df[x].value_counts()) > 1]\n",
    "            \n",
    "            if len(temp_df[\"gender\"].value_counts())>1:\n",
    "                temp_df = temp_df[['age_at_initial_pathologic_diagnosis',snp,\"gender\",\"PFI\",'PFI.time']+cov_col]\n",
    "            else:\n",
    "                temp_df = temp_df[['age_at_initial_pathologic_diagnosis',snp,\"PFI\",'PFI.time']+cov_col]\n",
    "                \n",
    "            #possibly use this line?\n",
    "            #temp_df=temp_df.drop(list(temp_df.var().index[temp_df.var()<0.01]),axis=1)\n",
    "            \n",
    "            temp_df=temp_df.fillna(0)\n",
    "            cph = CoxPHFitter(penalizer=0.01)\n",
    "            cph.fit(temp_df, event_col=\"PFI\", duration_col='PFI.time')\n",
    "            cancer_test.append(row[\"index\"])\n",
    "            cov_cols.append(cov_col)\n",
    "            df_coxph  = df_coxph.append(cph.summary.loc[[snp]])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8fc6c075-09bd-4f81-b5c2-d7d7ebe46102",
   "metadata": {},
   "outputs": [],
   "source": [
    "x=pd.DataFrame(zip(cancer_test,cov_cols))\n",
    "cov_col_table=x[~x[0].duplicated()]\n",
    "cov_col_table=cov_col_table.set_index(0)\n",
    "cov_col_table[\"index\"]=cov_col_table.index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "79ae781e-0098-415b-b5af-e6837c25e57c",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "#run coxph analysis on TRAIN datasets, use cov column from analysis on test datasets\n",
    "\n",
    "df_coxph_train = pd.DataFrame()\n",
    "cancer_train=[]\n",
    "\n",
    "for i,row in cov_col_table.iterrows():\n",
    "    print(row[\"index\"])\n",
    "    \n",
    "    for snp in rsid_lst:\n",
    "        temp_df = pfi_train[pfi_train[\"cancer\"]==row[\"index\"]]\n",
    "        temp_df=temp_df[~temp_df[snp].isnull()]\n",
    "        \n",
    "        cov_col=row[1]\n",
    "            \n",
    "        if max(temp_df[snp])==0:\n",
    "            continue\n",
    "        if len(temp_df[\"gender\"].value_counts())>1:\n",
    "            temp_df = temp_df[['age_at_initial_pathologic_diagnosis',snp,\"gender\",\"PFI\",'PFI.time']+cov_col]\n",
    "        else:\n",
    "            temp_df = temp_df[['age_at_initial_pathologic_diagnosis',snp,\"PFI\",'PFI.time']+cov_col] \n",
    "        \n",
    "        temp_df=temp_df.drop(list(temp_df.var().index[temp_df.var()<0.01]),axis=1) #seems to be necessary\n",
    "\n",
    "        temp_df=temp_df.fillna(0)\n",
    "        cph = CoxPHFitter(penalizer=0.01)\n",
    "        cph.fit(temp_df, event_col=\"PFI\", duration_col='PFI.time')\n",
    "        cancer_train.append(row[\"index\"])\n",
    "        df_coxph_train  = df_coxph_train.append(cph.summary.loc[[snp]])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "59d04369-c237-48d8-92c2-830c1fea3619",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Insert snp character column, add cancer index\n",
    "df_coxph[\"proxy_variant_chars\"]=df_coxph.index\n",
    "df_coxph_train[\"proxy_variant_chars\"]=df_coxph_train.index\n",
    "\n",
    "df_coxph[\"cancer\"]=cancer_test\n",
    "df_coxph_train[\"cancer\"]=cancer_train"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cb9dd065-80d9-4a3c-9f7c-091f76d23769",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Apply beta values from respective coxph dataset to respective surv dataset\n",
    "pfi_test_beta=pfi_test.copy()\n",
    "pfi_train_beta=pfi_train.copy()\n",
    "\n",
    "for i,row in df_coxph.iterrows():\n",
    "    if row[\"proxy_variant_chars\"] in pfi_test and not np.isnan(row[\"coef\"]):\n",
    "        pfi_test_beta.loc[pfi_test_beta[\"cancer\"]==row[\"cancer\"],i]=pfi_test.loc[pfi_test[\"cancer\"]==row[\"cancer\"],i]*row[\"coef\"]\n",
    "\n",
    "for i,row in df_coxph_train.iterrows():\n",
    "    if row[\"proxy_variant_chars\"] in pfi_train and not np.isnan(row[\"coef\"]):\n",
    "        pfi_train_beta.loc[pfi_train_beta[\"cancer\"]==row[\"cancer\"],i]=pfi_train[i][pfi_train[\"cancer\"]==row[\"cancer\"]]*row[\"coef\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "81323377-a681-4d07-abc8-af8c73b063ad",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Loop through lasso, drop snps, input list of hyperparams as well as respective cancer types\n",
    "#store coxph outcomes into final df for easy final plotting\n",
    "#run for loop by index and use 3 separate lists for input info?\n",
    "\n",
    "#### part 1, lasso model ####\n",
    "cancer_list=cov_col_table[\"index\"]\n",
    "cancer_list=cancer_list.drop(\"TGCT\")\n",
    "km_curve=\"off\"\n",
    "\n",
    "\n",
    "alphas=[0.0001,0.005,0.01,0.000005,0.001,0.0001,0.0001,0.01,0.00001,0.009,0.0001,0.0001,0.00001,0.00001,0.001,0.001]\n",
    "#.      BLCA.   BRCA. COAD     ESCA.   HNSC  KICH  KIRC.  KIRP.  LIHC  LUAD.   LUSC.  PAAD  READ. SKCM    STAD.  #THCA \n",
    "\n",
    "\n",
    "hazard_list=pd.DataFrame()\n",
    "all_survival_snps=pd.DataFrame()\n",
    "cohort_sizes_pfs=[]\n",
    "\n",
    "for i in range(len(cancer_list)):\n",
    "    \n",
    "    #Filter down dataframes to snps used in respective cancer type\n",
    "    surv_fdr_drop=surv_fdr[surv_fdr[\"cancer\"]==cancer_list[i]]\n",
    "    surv_fdr_drop=surv_fdr_drop[surv_fdr_drop[\"survival\"]==\"PFI\"]\n",
    "    surv_fdr_drop=surv_fdr_drop[surv_fdr_drop[\"logrank p\"]<0.05]\n",
    "    \n",
    "    cols_to_drop=np.setdiff1d(pfi_train_beta.filter(like=\":\").columns.values.tolist(),surv_fdr_drop[\"snps\"].to_list()).tolist()\n",
    "    \n",
    "    train_df=pfi_train_beta[pfi_train_beta[\"cancer\"]==cancer_list[i]]\n",
    "    test_df=pfi_test_beta[pfi_test_beta[\"cancer\"]==cancer_list[i]]\n",
    "    \n",
    "    train_df=train_df.drop(cols_to_drop,axis=1)\n",
    "    test_df=test_df.drop(cols_to_drop,axis=1)\n",
    "    \n",
    "    print(train_df.shape)\n",
    "    \n",
    "    train_y=train_df[\"PFI\"]\n",
    "    test_y=test_df[\"PFI\"]\n",
    "    \n",
    "    train_x=train_df.filter(like=\":\")\n",
    "    test_x=test_df.filter(like=\":\")\n",
    "\n",
    "    #filter using LASSO regression\n",
    "    # define mode\n",
    "    model = Lasso(alpha=alphas[i],random_state=random_state)\n",
    "    # fit model\n",
    "    model.fit(train_x, train_y)\n",
    "    p = model.predict(train_x)  # test the output by changing values\n",
    "\n",
    "    #Find model that gives best AUC, do a cutoff of weakest contributing SNPS, then run all analysis portions\n",
    "    preds=p\n",
    "    fpr, tpr, threshold = metrics.roc_curve(list(train_y), preds,pos_label=1)\n",
    "    roc_auc = metrics.auc(fpr, tpr)\n",
    "\n",
    "    print(roc_auc)\n",
    "\n",
    "    #### part 2 ####\n",
    "\n",
    "    cohort_sizes_pfs.append([cancer_list[i],test_x.shape[0]])\n",
    "    #remake SNP list with dropped \"useless\" SNPs\n",
    "    \n",
    "    print(\"SNPs dropped = \",len(train_x.columns.values[model.coef_==0].tolist()),\"out of \",len(train_x.columns.values))\n",
    "    train_df=train_df.drop(train_x.columns.values[model.coef_==0].tolist(),axis=1)\n",
    "    test_df=test_df.drop(test_x.columns.values[model.coef_==0].tolist(),axis=1)\n",
    "    \n",
    "    surv_fdr_drop=surv_fdr_drop[~surv_fdr_drop[\"snps\"].isin(test_x.columns.values[model.coef_==0].tolist())]\n",
    "    all_survival_snps=all_survival_snps.append(surv_fdr_drop)\n",
    "    \n",
    "    train_x=train_df.filter(like=\":\")\n",
    "    test_x=test_df.filter(like=\":\")\n",
    "    \n",
    "    regressor_xgb = xgboost.XGBRFRegressor(n_estimators = 300, random_state = random_state,learning_rate=0.5,max_depth=9)\n",
    "    regressor_xgb.fit(train_x, train_y)  \n",
    "\n",
    "    p = regressor_xgb.predict(test_x)  # test the output by changing values\n",
    "\n",
    "    test_df[\"LASSO_burden\"]=p*100\n",
    "    cat1 = test_df[test_df[\"PFI\"]==0]\n",
    "    cat2 = test_df[test_df[\"PFI\"]==1]\n",
    "    print(\"mannWhitnetPredScore=\",stats.mannwhitneyu(cat1[\"LASSO_burden\"], cat2[\"LASSO_burden\"])[1])\n",
    "\n",
    "    #now do hazard plot on PRS scores generated...\n",
    "    test_df_coxPH=test_df[list(np.append(['Stage II', 'Stage III', 'Stage I','ajcc_pathologic_tumor_stage',\n",
    "            'PFI', 'PFI.time',\n",
    "           'age_at_initial_pathologic_diagnosis',\n",
    "           'LASSO_burden'],cov_col_table.loc[cancer_list[i],1]))]\n",
    "    \n",
    "    #filter covariates that throw errors due to low variance\n",
    "    test_df_coxPH=test_df_coxPH.drop(list(test_df_coxPH.var().index[test_df_coxPH.var()<0.01]),axis=1) #seems to be necessary\n",
    "    test_df_coxPH=test_df_coxPH.fillna(0)\n",
    "\n",
    "    cph = CoxPHFitter(penalizer=0.01)\n",
    "\n",
    "    cph.fit(test_df_coxPH, duration_col = 'PFI.time', event_col = 'PFI')\n",
    "    print(cancer_list[i])\n",
    "    print(cph.summary.loc[\"LASSO_burden\"])\n",
    "    \n",
    "    hazard_list=hazard_list.append(cph.summary.loc[\"LASSO_burden\"])\n",
    "    \n",
    "    if km_curve==\"on\":\n",
    "        #plot KM curve\n",
    "        surv_df_canc=test_df_coxPH\n",
    "\n",
    "        fig=plt.figure(figsize=(5,5))\n",
    "        ax1 = plt.subplot(1,1,1)\n",
    "\n",
    "        quantiles=np.quantile(surv_df_canc[\"LASSO_burden\"],[0,0.25,0.75,1])\n",
    "        surv_df_canc['group'] = pd.cut(surv_df_canc[\"LASSO_burden\"], quantiles, labels=[0,1,2])\n",
    "\n",
    "        kmf = KaplanMeierFitter()\n",
    "        kmf.fit(surv_df_canc[surv_df_canc['group']==2]['PFI.time'], surv_df_canc[surv_df_canc['group']==2]['PFI'],label=\"High, N=\"+str(surv_df_canc[surv_df_canc['group']==2].shape[0]))\n",
    "        kmf.plot(ci_show=False, ax=ax1)\n",
    "        kmf.fit(surv_df_canc[surv_df_canc['group']==1]['PFI.time'], surv_df_canc[surv_df_canc['group']==1]['PFI'],label=\"Med, N=\"+str(surv_df_canc[surv_df_canc['group']==1].shape[0]))\n",
    "        kmf.plot(ci_show=False, ax=ax1)\n",
    "        kmf.fit(surv_df_canc[surv_df_canc['group']==0]['PFI.time'], surv_df_canc[surv_df_canc['group']==0]['PFI'],label=\"Low, N=\"+str(surv_df_canc[surv_df_canc['group']==0].shape[0]))\n",
    "        kmf.plot(ci_show=False, ax=ax1)\n",
    "\n",
    "        results = logrank_test(surv_df_canc[surv_df_canc[\"group\"]==2]['PFI.time'], surv_df_canc[surv_df_canc[\"group\"]==0]['PFI.time'],event_observed_A=surv_df_canc[surv_df_canc[\"group\"]==2]['PFI'], event_observed_B=surv_df_canc[surv_df_canc[\"group\"]==0]['PFI'], alpha=.95)\n",
    "        results2 = logrank_test(surv_df_canc[surv_df_canc[\"group\"]==2]['PFI.time'], surv_df_canc[surv_df_canc[\"group\"]==1]['PFI.time'],event_observed_A=surv_df_canc[surv_df_canc[\"group\"]==2]['PFI'], event_observed_B=surv_df_canc[surv_df_canc[\"group\"]==1]['PFI'], alpha=.95)\n",
    "        results3 = logrank_test(surv_df_canc[surv_df_canc[\"group\"]==1]['PFI.time'], surv_df_canc[surv_df_canc[\"group\"]==0]['PFI.time'],event_observed_A=surv_df_canc[surv_df_canc[\"group\"]==1]['PFI'], event_observed_B=surv_df_canc[surv_df_canc[\"group\"]==0]['PFI'], alpha=.95)\n",
    "\n",
    "        plt.ylabel('% survival')\n",
    "        plt.xlabel('PFI time (days)')\n",
    "        plt.title(str(cancer_list[i])+\"\\n\"+\"High vs Low \"+str(np.round(results.p_value,4))+\"\\n\"+\"High vs Med \"+str(np.round(results2.p_value,4))+\"\\n\"+\"Med vs Low \"+str(np.round(results3.p_value,4)))\n",
    "        plt.legend(frameon=False)\n",
    "        plt.savefig(\"../plots/fig4/\"+cancer_list[i]+\"_0.25_0.75_PFI_KM.pdf\")\n",
    "        \n",
    "#all_survival_snps.to_csv(\"/cellar/users/tsears/projects/germline-immune/data/fig6/PFS_input_snps.txt\",sep=\"\\t\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f8976339-4d27-4955-abd1-6318528647b5",
   "metadata": {},
   "outputs": [],
   "source": [
    "haz_plot_=hazard_list\n",
    "haz_plot_.index=cancer_list\n",
    "haz_plot_[\"Cancer Type\"]=cancer_list\n",
    "#haz_plot_=haz_plot_.drop(haz_plot_.index[abs(haz_plot_[\"coef\"])>1][0])\n",
    "haz_plot_=haz_plot_.drop([\"KICH\"])\n",
    "haz_plot_=fdr(haz_plot_,False,\"p\")\n",
    "haz_plot_=haz_plot_.sort_values(by=\"exp(coef)\",ascending=True)\n",
    "plt.figure(figsize=(15,3))\n",
    "plt.errorbar(x=haz_plot_[\"Cancer Type\"].tolist(),y=haz_plot_[\"exp(coef)\"].tolist(),yerr=haz_plot_[\"se(coef)\"].tolist(),fmt=\"o\")\n",
    "haz_plot_thresh=haz_plot_[haz_plot_[\"fdr\"]<0.05]\n",
    "plt.errorbar(x=haz_plot_thresh[\"Cancer Type\"].tolist(),y=haz_plot_thresh[\"exp(coef)\"].tolist(),yerr=haz_plot_thresh[\"se(coef)\"].tolist(),fmt=\"o\",color=\"red\")\n",
    "plt.xlabel(\"Cancer Type\")\n",
    "plt.ylabel(\"Hazard Ratio\")\n",
    "plt.axhline(1,linestyle=\"--\",color=\"gray\")\n",
    "plt.savefig(\"../plots/cancer_risk_PFI_hazardV2_long.pdf\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cc6d3af5-9070-4b9e-a541-a94b21715d0f",
   "metadata": {},
   "outputs": [],
   "source": [
    "cohort_sizes_pfs"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "germtime",
   "language": "python",
   "name": "germtime"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
