{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "a96c36f5-6da8-4e32-8bad-d99cd8ed49b6",
   "metadata": {},
   "source": [
    "### Cancer Survival Experiment PFS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "5e9d9da3-393c-40e8-b703-8bb8afb3c643",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import statsmodels.stats.multitest as multi\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "1795a962-a973-41d9-a11d-d6ce9ece69f8",
   "metadata": {},
   "outputs": [],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import matplotlib\n",
    "matplotlib.rcParams['pdf.fonttype'] = 42\n",
    "matplotlib.rcParams['ps.fonttype'] = 42"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "72faf8bd-e861-43d0-9242-3a3263748d30",
   "metadata": {},
   "outputs": [],
   "source": [
    "import lifelines\n",
    "from lifelines import CoxPHFitter\n",
    "from lifelines import KaplanMeierFitter\n",
    "import statsmodels.stats.multitest as multi\n",
    "from sklearn.linear_model import LogisticRegression\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "from lifelines.statistics import logrank_test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "33b52ca3-5c09-488f-9c4a-aae82e0f5aae",
   "metadata": {},
   "outputs": [],
   "source": [
    "from scipy.stats import mannwhitneyu\n",
    "import statsmodels.formula.api as smf\n",
    "from scipy.stats import mannwhitneyu\n",
    "from scipy import stats"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "65dc6762-1cea-4d60-8f4e-e2613340a817",
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "with open('/cellar/users/mpagadal/resources/tcga/ensembl_map.json', 'r') as f:\n",
    "    ensembl = json.load(f)\n",
    "ensembl={k.split(\".\")[0]:v for k,v in ensembl.items()}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "e5317a7a-3c3e-4fcb-bfab-cc844970df59",
   "metadata": {},
   "outputs": [],
   "source": [
    "random_state=75"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "ce8a7338-29a4-4209-a900-aec11bb35b27",
   "metadata": {},
   "outputs": [],
   "source": [
    "def make_surv_df(surv,surv_type,raw):\n",
    "    \n",
    "    '''\n",
    "    make TCGA survival dataframe\n",
    "    '''\n",
    "    \n",
    "    #get survival dataframe\n",
    "    surv=pd.read_csv(surv,index_col=0)\n",
    "    surv=surv.rename(columns={\"bcr_patient_barcode\":\"FID\"})\n",
    "    surv=surv[surv[\"Redaction\"].isnull()]\n",
    "    \n",
    "    #get phenos\n",
    "    raw=pd.read_csv(raw,delim_whitespace=True,index_col=0)\n",
    "    raw=raw.iloc[:,5:]\n",
    "    raw=raw.reset_index()\n",
    "    raw=raw.rename(columns={\"index\":\"FID\"})\n",
    "\n",
    "    #combine dataframes\n",
    "    full_surv=pd.merge(surv[[\"FID\",surv_type,surv_type+\".time\",\"age_at_initial_pathologic_diagnosis\",\"race\",\"gender\",\"ajcc_pathologic_tumor_stage\"]], raw, on=\"FID\", how=\"left\")\n",
    "    #full_surv=full_surv[full_surv[surv_type+\".time\"]<1825] #censor at 5 years\n",
    "    #get patients with Stage designations\n",
    "    full_surv=full_surv[full_surv[\"ajcc_pathologic_tumor_stage\"].str.contains(\"Stage\")]\n",
    "    full_surv=full_surv[~(full_surv[\"ajcc_pathologic_tumor_stage\"].isin([\"Stage X\"]))]\n",
    "    #split A,B,C from stage designations\n",
    "    full_surv[\"ajcc_pathologic_tumor_stage\"]=full_surv[\"ajcc_pathologic_tumor_stage\"].str.split(\"A\").str[0]\n",
    "    full_surv[\"ajcc_pathologic_tumor_stage\"]=full_surv[\"ajcc_pathologic_tumor_stage\"].str.split(\"B\").str[0]\n",
    "    full_surv[\"ajcc_pathologic_tumor_stage\"]=full_surv[\"ajcc_pathologic_tumor_stage\"].str.split(\"C\").str[0]\n",
    "    print(full_surv.shape)\n",
    "    full_surv=full_surv.join(pd.get_dummies(full_surv[\"ajcc_pathologic_tumor_stage\"]))\n",
    "    #remap gender\n",
    "    full_surv[\"gender\"]=full_surv[\"gender\"].map({\"MALE\":0,\"FEMALE\":1})\n",
    "    \n",
    "    full_surv=full_surv[~full_surv[surv_type].isnull()]\n",
    "    full_surv=full_surv[~full_surv[surv_type+\".time\"].isnull()]\n",
    "    \n",
    "    full_surv=full_surv[~full_surv[\"age_at_initial_pathologic_diagnosis\"].isnull()]\n",
    "    \n",
    "    #map to cancer type\n",
    "    canc = pd.read_csv(\"/cellar/controlled/dbgap-genetic/phs000178_TCGA/birdseed-processing/gtype.meta\", delimiter=\"\\t\", header=None)\n",
    "    mp = canc.set_index(2)[0].to_dict()\n",
    "    full_surv[\"cancer\"]=full_surv[\"FID\"].map(mp)\n",
    "    full_surv=full_surv[~full_surv[\"cancer\"].isnull()]\n",
    "    \n",
    "    full_surv['ajcc_pathologic_tumor_stage'] = full_surv['ajcc_pathologic_tumor_stage'].replace({'Stage 0':0,'Stage I': 1,'Stage II': 2,'Stage III': 3,'Stage IV': 4})\n",
    "    \n",
    "    return(full_surv)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "22914dd2-1601-4603-8f0c-d22ccc7a2fad",
   "metadata": {},
   "outputs": [],
   "source": [
    "def run_surv_plot(df_annot,surv_df,surv_type,out):\n",
    "    \n",
    "    '''\n",
    "    plot kaplan meier for each snp in df_annot\n",
    "    '''\n",
    "    for i,row in df_annot.iterrows():\n",
    "        try:\n",
    "            surv_df_canc=surv_df[surv_df[\"cancer\"]==row[\"cancer\"]]\n",
    "            x=[x for x in surv_df.columns if row[\"snps\"].rsplit(\":\",2)[0] in x][0]\n",
    "            rs=x.rsplit(\":\",2)[0]\n",
    "        \n",
    "            allele1=x.split(\":\")[2]\n",
    "            allele2=str(x.split(\"_\")[0]).split(\":\")[3]\n",
    "            minor=x.split(\"_\")[1]\n",
    "            if allele1 == minor:\n",
    "                major=allele2\n",
    "            else:\n",
    "                major=allele1\n",
    "            \n",
    "            fig=plt.figure(figsize=(12,5))\n",
    "            ax1 = plt.subplot(1,2,1)\n",
    "            \n",
    "            groups = surv_df_canc[x]\n",
    "            ix0 = (groups == 0)\n",
    "            ix1 = (groups == 1)\n",
    "            ix2 = (groups == 2)\n",
    "\n",
    "            kmf = KaplanMeierFitter()\n",
    "            kmf.fit(surv_df_canc[surv_type+'.time'][ix2], surv_df_canc[surv_type][ix2],label=minor+minor)\n",
    "            kmf.plot(color='green', ci_show=False, ax=ax1)\n",
    "            \n",
    "            kmf.fit(surv_df_canc[surv_type+'.time'][ix1], surv_df_canc[surv_type][ix1],label=major+minor)\n",
    "            kmf.plot(ax=ax1, color='orange', ci_show=False)\n",
    "    \n",
    "            kmf.fit(surv_df_canc[surv_type+'.time'][ix0], surv_df_canc[surv_type][ix0],label=major+major)\n",
    "            kmf.plot(ax=ax1, color='blue', ci_show=False)\n",
    "\n",
    "            results = logrank_test(surv_df_canc[surv_type+'.time'][ix0], surv_df_canc[surv_type+'.time'][ix2],event_observed_A=surv_df_canc[surv_type][ix0], event_observed_B=surv_df_canc[surv_type][ix2], alpha=.95) \n",
    "            plt.title('{}\\n{}\\nLog-rank test: p<{:.3}'.format(x,row[\"cancer\"],results.p_value))\n",
    "\n",
    "            plt.ylabel('% survival')\n",
    "            plt.xlabel(surv_type+' time (days)')\n",
    "            plt.legend(frameon=False)\n",
    "\n",
    "            plt.tight_layout()\n",
    "\n",
    "            plt.savefig(out+surv_type+\".\"+x+\".\"+row[\"cancer\"]+'.pdf')\n",
    "            plt.close()\n",
    "        except:\n",
    "            pass\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "890eb730-3d4c-48f3-a6bf-fcb1e42ffe77",
   "metadata": {},
   "outputs": [],
   "source": [
    "def fdr(df,pseudo,p):\n",
    "    \n",
    "    '''\n",
    "    calculate Benjamini-Hochberg FDR\n",
    "    pseudo: True or False\n",
    "    '''\n",
    "    \n",
    "    #add pseudocounta\n",
    "    if pseudo == True:\n",
    "        df[p]=df[p]+0.000000000001\n",
    "    \n",
    "    #calculate benjamin-hochberg fdr\n",
    "    probabilities = df[p].to_numpy()\n",
    "    report = multi.multipletests(probabilities,\n",
    "                alpha=0.05,\n",
    "                method=\"fdr_bh\",\n",
    "                is_sorted=False,\n",
    "            )\n",
    "    discoveries = report[1]\n",
    "    df[\"fdr\"]=discoveries\n",
    "    df[\"fdr\"]=pd.to_numeric(df[\"fdr\"])\n",
    "    df[\"-log10(fdr)\"]=-np.log10(df[\"fdr\"])\n",
    "    \n",
    "    return(df)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bd945b3c-0d15-403f-b5ca-a5231e8c19b7",
   "metadata": {},
   "source": [
    "### Get associations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "bebc0f56-7824-4264-a14e-0725703cf2eb",
   "metadata": {},
   "outputs": [],
   "source": [
    "snps=pd.read_csv(\"/cellar/users/mpagadal/projects/germline-immune/data/supplemental/Supplemental_Table_3.csv\",delimiter=\"\\t\")\n",
    "snps[\"snp_noallele\"]=snps[\"proxy_variant\"].str.rsplit(\":\",2).str[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "fed7ddf4-82e6-41e6-bfa8-dc4d11f1708e",
   "metadata": {},
   "outputs": [],
   "source": [
    "fdr_thresh=0.05"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "5ab28e35-c541-451d-9567-a3c1f05eacd3",
   "metadata": {},
   "outputs": [],
   "source": [
    "surv_fdr=pd.read_csv(\"/cellar/users/mpagadal/projects/germline-immune/data/supplemental/Supplemental_Table_6.csv\",index_col=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "9923f94d-8b85-4662-919e-56d6696fc64c",
   "metadata": {},
   "outputs": [],
   "source": [
    "prog_snps=surv_fdr[surv_fdr[\"P\"]<fdr_thresh][\"ID\"].unique().tolist()\n",
    "prog_snps=[x.split(\"_\")[0] for x in prog_snps]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "c91731ea-d95b-4408-abf3-ac746ca1db49",
   "metadata": {},
   "outputs": [],
   "source": [
    "#check how many snps are associated with survival in multiple cancer types\n",
    "surv_number=surv_fdr[surv_fdr[\"P\"]<fdr_thresh][[\"ID\",\"cancer\"]].drop_duplicates().groupby([\"ID\"]).size().reset_index()\n",
    "surv_number=surv_number.sort_values(by=0,ascending=False)\n",
    "surv_number[\"snp_noallele\"]=surv_number[\"ID\"].str.rsplit(\":\",2).str[0]\n",
    "snp_sig=snps[snps[\"snp_noallele\"].isin(surv_number[surv_number[0]>1][\"snp_noallele\"].tolist())]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "786ac172-daf4-47a0-8733-424b009f1393",
   "metadata": {},
   "outputs": [],
   "source": [
    "snp_sig=snp_sig.loc[~snp_sig[\"proxy_variant\"].duplicated()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "5eac69ee-f37e-441c-87ab-1dfbb709977f",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3457: DtypeWarning: Columns (8,9,27,42,44,45,46,47,48,49,50,51,52,53,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,161,162,163,164,165,166,167,168,169,170,171,172,173,174,175,176,177,178,179,180,181,182,183,184,185,186,187,188,189,190,191,192,193,194,195,196,197,198,199,200,201,202,203,204,205,206,207,208,209,210,211,212,213,214,215,216,217,218,219,220,221,222,223,224,225,226,227,228,229,230,231,232,233,234,235,236,237,238,239,240,241,242,243,244,245,246,247,248,249,250,251,252,253,254,255,256,257,258,259,260,261,262,263,264,265,266,267,268,269,270,271,272,273,274,275,276,277,278,279,280,281,282,283,284,285,286,287,288,289,290,291,292,293,294,295,296,297,298,299,300,301,302,303,304,305,306,307,308,309,310,311,312,313,314,315,316,317,318,319,320,321,322,323,324,325,326,327,328,329,330,333,334,335,336,338,339,340,341,342,343,344,345,346,347,348,349,350,351,352,353,354,355,356,357,358,359,360,361,362,363,364,365,366,367,368,369,370,371,372,373,374,375,376,377,378,379,380,381,382,383,384,385,386,387,388,389,390,391,392,393,394,395,396,397,398,399,400,401,402,403,404,405,406,407,408,409,410,411,412,413,414,415,416,417,418,419,420,421,422,423,424,425,426,427,428,429,430,431,432,433,434,435,436,437,438,439,440,441,442,443,444,445,446,447,448,449,450,451,452,453,454,455,456,457,458,459,460,461,462,463,464,465,466,467,468,469,470,471,472,473,474,475,476,477,478,479,480,481,482,483,484,485,486,487,488,489,490,491,492,493,494,495,496,497,498,499,500,501,502,503,504,505,506,507,508,509,510,511,512,513,514,515,516,517,518,519,520,521,522,523,524,525,526,527,528,529,530,531,532,533,534,535,536,537,538,539,540,541,542,543,544,545,546,547,548,549,550,551,552,553,554,555,556,557,558,559,560,561,562,563,564,565,566,567,568,569,570,571,572,573,574,575,576,577,578,579,580,581,582,583,584,585,586,587,588,589,590,591,592,593,594,595,596,597,598,599,600,601,602,603,604,605,606,607,608,609,610,611,612,613,614,615,616,617,618,619,620,621,622,623,624,625,626,627,628,629,630,631,632,633,634,635,636,637,639,640,641,642,643,644,645,646,647,648,649,650,651,652,653,654,655,656,657,658,659,660,661,662,663,664,665,666,667,668,669,670,671,672,673,674,675,676,677,678,679,680,681,682,683,684,685,686,687,688,689,690,691,692,693,694,695,696,697,698,699,700,701,702,703,704,705,706,707,708,709,710,711,712,713,714,715,716,717,718,719,720,721,722,723,724,725,726,727,728,729,730,731,732,733,734,735,736,737,738,739,740,741,742,743,744) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  exec(code_obj, self.user_global_ns, self.user_ns)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "history_of_neoadjuvant_treatment\n",
      "radiation_therapy\n",
      "pathologic_T\n",
      "pathologic_M\n",
      "pathologic_N\n",
      "pathologic_stage\n",
      "lymph_node_examined_count\n",
      "number_of_lymphnodes_positive_by_he\n",
      "neoplasm_histologic_grade\n",
      "height\n",
      "weight\n",
      "tobacco_smoking_history\n",
      "targeted_molecular_therapy\n",
      "menopause_status\n",
      "breast_carcinoma_progesterone_receptor_status\n",
      "breast_carcinoma_estrogen_receptor_status\n",
      "lab_proc_her2_neu_immunohistochemistry_receptor_status\n",
      "lab_procedure_her2_neu_in_situ_hybrid_outcome_type\n",
      "diagnosis\n",
      "alcohol_history_documented\n",
      "synchronous_colon_cancer_present\n",
      "hypertension\n",
      "seizure_history\n",
      "history_ionizing_rt_to_head\n",
      "headache_history\n",
      "extrathyroid_carcinoma_present_extension_status\n",
      "gleason_score\n",
      "diagnosis_subtype\n",
      "history_hepato_carcinoma_risk_factor\n",
      "ct_scan\n",
      "history_of_undescended_testis\n",
      "history_hypospadias\n",
      "history_myasthenia_gravis\n",
      "mitotane_therapy\n",
      "ebv_antibody_status\n",
      "relative_family_cancer_history\n",
      "hist_hepato_carc_fact\n",
      "creatinine_value_in_mg_dl\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/pandas/core/indexing.py:1817: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self._setitem_single_column(loc, value, pi)\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:52: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:55: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:56: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:57: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:58: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:59: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:60: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:61: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:62: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "pathologic_T\n",
      "pathologic_M\n",
      "pathologic_N\n",
      "neoplasm_histologic_grade\n",
      "tobacco_smoking_history\n",
      "menopause_status\n",
      "extrathyroid_carcinoma_present_extension_status\n",
      "diagnosis_subtype\n",
      "history_hepato_carcinoma_risk_factor\n",
      "hist_hepato_carc_fact\n"
     ]
    }
   ],
   "source": [
    "surv=pd.read_csv(\"clinical_PANCAN_patient_with_followup.tsv\",delimiter=\"\\t\",encoding=\"unicode_escape\")\n",
    "keep=pd.read_csv(\"surv_var_keep_MP.csv\") #keep only curated survival variables\n",
    "surv_clean=surv[[\"acronym\"]+keep[\"var\"].tolist()]\n",
    "\n",
    "total_counts=pd.DataFrame()\n",
    "for x in surv_clean.columns[2:]:\n",
    "    surv_not_null=surv_clean[~surv_clean[x].isnull()]\n",
    "    print(x)\n",
    "    try:\n",
    "        surv_not_null=surv_not_null[~(surv_not_null[x].str.contains(\"Not Av\"))]\n",
    "        surv_not_null=surv_not_null[~(surv_not_null[x].str.contains(\"Not Ap\"))]\n",
    "        surv_not_null=surv_not_null[~(surv_not_null[x].str.contains(\"Unk\"))]\n",
    "        df_counts=surv_not_null[\"acronym\"].value_counts().reset_index()\n",
    "        df_counts[\"var\"]=x\n",
    "        total_counts=total_counts.append(df_counts)\n",
    "    except:\n",
    "        df_counts=surv_not_null[\"acronym\"].value_counts().reset_index()\n",
    "        df_counts[\"var\"]=x\n",
    "        total_counts=total_counts.append(df_counts)\n",
    "supp_table=total_counts.groupby('index')['var'].apply(lambda x: ','.join(x)).reset_index()\n",
    "supp_table.loc[14][\"var\"]='history_of_neoadjuvant_treatment,radiation_therapy,height,weight,T2_pathologic_T,T3_pathologic_T,T4_pathologic_T,M1_pathologic_M,N1_pathologic_N,G1_neoplasm_histologic_grade,G2_neoplasm_histologic_grade,G3_neoplasm_histologic_grade,G4_neoplasm_histologic_grade,alcohol,NAFLD,hepatitis'\n",
    "supp_table.loc[15][\"var\"]='radiation_therapy,targeted_molecular_therapy,T2_pathologic_T,T3_pathologic_T,T4_pathologic_T,M1_pathologic_M,N1_pathologic_N,N2_pathologic_N,N3_pathologic_N,Current smoker_tobacco_smoking_history,Lifelong Non-smoker_tobacco_smoking_history'\n",
    "supp_table.loc[26][\"var\"]='radiation_therapy,history_of_undescended_testis,T2_pathologic_T,M1_pathologic_M,N1_pathologic_N,N2_pathologic_N'\n",
    "supp_table.loc[29][\"var\"]='radiation_therapy,height,weight,hypertension,G2_neoplasm_histologic_grade,G3_neoplasm_histologic_grade,High Grade_neoplasm_histologic_grade,Peri (6-12 months since last menstrual period)_menopause_status,Post (prior bilateral ovariectomy OR >12 mo since LMP with no prior hysterectomy)_menopause_status,Pre (<6 months since LMP AND no prior bilateral ovariectomy AND not on estrogen replacement)_menopause_status'\n",
    "\n",
    "for x in surv_clean.columns:\n",
    "    try:\n",
    "        m = surv_clean[x].str.contains('\\[', na=False)\n",
    "        surv_clean.loc[m,x] = np.nan\n",
    "    except:\n",
    "        pass\n",
    "\n",
    "# map variables to category\n",
    "mp_map={'Equivocal':np.nan,\n",
    " 'Indeterminate':np.nan,\n",
    " 'NO':0,\n",
    " 'Negative':0,\n",
    " 'No':0,\n",
    " 'Not Performed':np.nan,\n",
    " 'Positive':1,\n",
    " 'YES':1,\n",
    " 'Yes':1,\n",
    " 'Yes, Pharmaceutical Treatment Prior to Resection':1,\n",
    " 'Yes, Radiation Prior to Resection':1,\n",
    " 'Yes; bilateral':1,\n",
    " 'Yes; left testicle only':1,\n",
    " 'Yes; right testicle only':1,\n",
    " 'negative':0,\n",
    " 'positive':1,\n",
    "}\n",
    "for var in keep[keep[\"coding\"]==\"map\"][\"var\"].unique():\n",
    "    surv_clean[var]=surv_clean[var].map(mp_map)  \n",
    "\n",
    "#clean staging\n",
    "surv_clean[\"pathologic_T\"]=surv_clean[\"pathologic_T\"].str[0:2]\n",
    "surv_clean[\"pathologic_N\"]=surv_clean[\"pathologic_N\"].str[0:2]\n",
    "surv_clean[\"pathologic_M\"]=surv_clean[\"pathologic_M\"].str[0:2]\n",
    "surv_clean[\"pathologic_stage\"]=surv_clean[\"pathologic_stage\"].str.replace(\"A\",\"\")\n",
    "surv_clean[\"pathologic_stage\"]=surv_clean[\"pathologic_stage\"].str.replace(\"B\",\"\")\n",
    "surv_clean[\"pathologic_stage\"]=surv_clean[\"pathologic_stage\"].str.replace(\"C\",\"\")\n",
    "surv_clean[\"pathologic_stage\"]=surv_clean[\"pathologic_stage\"].replace(\"Stage X\",np.nan)\n",
    "surv_clean['pathologic_stage'] = surv_clean['pathologic_stage'].replace({'Stage 0':0,'Stage I': 1,'Stage II': 2,'Stage III': 3,'Stage IV': 4,'Stage IS':0,'I or II NOS':0})\n",
    "\n",
    "#dummy code specified variables\n",
    "mp_dummy={}\n",
    "for x in keep[keep[\"coding\"]==\"dummy\"][\"var\"].unique():\n",
    "    print(x)\n",
    "    df_dummy=pd.get_dummies(surv_clean[x])\n",
    "    df_dummy.columns=[col+\"_\"+x for col in df_dummy.columns]\n",
    "    mp_dummy[x]=df_dummy.columns.tolist()\n",
    "    surv_clean=surv_clean.join(df_dummy)\n",
    "    del surv_clean[x]\n",
    "    if \"None\" in surv_clean.columns:\n",
    "        del surv_clean[\"None\"]\n",
    "    if \"Other\" in surv_clean.columns:\n",
    "        del surv_clean[\"Other\"]\n",
    "    \n",
    "surv=pd.read_csv(\"/cellar/users/mpagadal/resources/from-labmembers/andrea/Liu2018.TCGA_survival.csv\")\n",
    "surv=surv[[\"bcr_patient_barcode\",\"gender\",\"age_at_initial_pathologic_diagnosis\",\"type\",\"OS\",\"OS.time\",\"PFI\",\"PFI.time\"]]\n",
    "surv=pd.merge(surv,surv_clean,on=\"bcr_patient_barcode\",how=\"left\")\n",
    "surv[\"gender\"]=surv[\"gender\"].map({\"MALE\":0,\"FEMALE\":1})\n",
    "\n",
    "del surv[\"T0_pathologic_T\"]\n",
    "del surv[\"TX_pathologic_T\"]\n",
    "del surv[\"Ti_pathologic_T\"]\n",
    "del surv[\"T1_pathologic_T\"]\n",
    "\n",
    "del surv[\"MX_pathologic_M\"]\n",
    "del surv[\"cM_pathologic_M\"]\n",
    "del surv[\"M0_pathologic_M\"]\n",
    "\n",
    "del surv[\"N0_pathologic_N\"]\n",
    "del surv[\"NX_pathologic_N\"]\n",
    "\n",
    "del surv[\"Indeterminate (neither Pre or Postmenopausal)_menopause_status\"]\n",
    "del surv[\"No History of Primary Risk Factors_hist_hepato_carc_fact\"]\n",
    "del surv[\"Other_hist_hepato_carc_fact\"]\n",
    "del surv[\"lymph_node_examined_count\"]\n",
    "del surv[\"synchronous_colon_cancer_present\"]\n",
    "del surv[\"pathologic_stage\"]\n",
    "\n",
    "del surv[\"GX_neoplasm_histologic_grade\"]\n",
    "del surv[\"G1_neoplasm_histologic_grade\"]\n",
    "\n",
    "del surv[\"Current Reformed Smoker, Duration Not Specified_tobacco_smoking_history\"]\n",
    "del surv[\"Current reformed smoker for < or = 15 years_tobacco_smoking_history\"]\n",
    "del surv[\"Current reformed smoker for > 15 years_tobacco_smoking_history\"]\n",
    "del surv[\"history_of_neoadjuvant_treatment\"]\n",
    "\n",
    "surv[\"alcohol\"]=surv[[x for x in surv.columns if \"Alcohol consumption\" in x]].sum(axis=1)\n",
    "surv[\"NAFLD\"]=surv[[x for x in surv.columns if \"Nonalcoholic Fatty Liver Diseas\" in x]].sum(axis=1)\n",
    "surv[\"hepatitis\"]=surv[[x for x in surv.columns if \"Hepatitis\" in x]].sum(axis=1)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "572b565a-5c60-499f-9c88-9e45bcb160d3",
   "metadata": {},
   "outputs": [],
   "source": [
    "#merge snps onto clinical file\n",
    "raw=pd.read_csv(\"/cellar/users/mpagadal/projects/germline-immune/data/genotypes/all.time.raw\",delim_whitespace=True)\n",
    "raw=raw.rename(columns={\"FID\":\"bcr_patient_barcode\"})\n",
    "surv=pd.merge(surv,raw,on=\"bcr_patient_barcode\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "936c6712-5d13-447b-93d9-712682f2c7b5",
   "metadata": {},
   "outputs": [],
   "source": [
    "#get snps\n",
    "prog_snps=surv_fdr[surv_fdr[\"P\"]<fdr_thresh][\"ID\"].unique().tolist()\n",
    "prog_snps=[x.split(\"_\")[0] for x in prog_snps]\n",
    "\n",
    "rsid_lst=[]\n",
    "for snp in prog_snps:\n",
    "    try:\n",
    "        rsid_lst.append([x for x in surv.columns if snp.rsplit(\":\",2)[0] in x][0])\n",
    "    except:\n",
    "        print(snp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "9e9713cd-dd9b-4acc-b605-1f7bcb5b7e04",
   "metadata": {},
   "outputs": [],
   "source": [
    "canc_counts=surv_clean[\"acronym\"].value_counts().reset_index()\n",
    "canc_keep=canc_counts[canc_counts[\"acronym\"]>100][\"index\"].tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "150b0764-503d-4f98-a784-348f857a2f47",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(6875, 976)\n",
      "(6875, 976)\n"
     ]
    }
   ],
   "source": [
    "#Create OS surv dataframe and split 70/30 by each cancer type\n",
    "\n",
    "os_surv=make_surv_df(\"/cellar/users/mpagadal/resources/from-labmembers/andrea/Liu2018.TCGA_survival.csv\",\"OS\",\"/cellar/users/mpagadal/projects/germline-immune/data/genotypes/all.time.raw\")\n",
    "pfi_surv=make_surv_df(\"/cellar/users/mpagadal/resources/from-labmembers/andrea/Liu2018.TCGA_survival.csv\",\"PFI\",\"/cellar/users/mpagadal/projects/germline-immune/data/genotypes/all.time.raw\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "f594c35d-ae40-4035-8db0-d228488ed105",
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.model_selection import train_test_split\n",
    "\n",
    "os_test=pd.DataFrame()\n",
    "os_train=pd.DataFrame()\n",
    "\n",
    "for cancer in np.unique(pfi_surv[\"cancer\"]):\n",
    "    \n",
    "    temp_train,temp_test=train_test_split(pfi_surv[pfi_surv[\"cancer\"]==cancer],test_size=0.3,random_state=random_state)\n",
    "    os_test=os_test.append(temp_test)\n",
    "    os_train=os_train.append(temp_train)\n",
    "\n",
    "surv_filt=surv.drop([\"gender\",\"age_at_initial_pathologic_diagnosis\",\"type\",\"OS\",\"OS.time\",\"PFI\",\"PFI.time\",\"acronym\"],axis=1)\n",
    "surv_filt=surv_filt.drop(surv_filt.filter(like=\":\").columns.values,axis=1)\n",
    "\n",
    "#merge additional covariates into df\n",
    "os_test=pd.merge(os_test,surv_filt,left_on=\"FID\",right_on=\"bcr_patient_barcode\")\n",
    "os_train=pd.merge(os_train,surv_filt,left_on=\"FID\",right_on=\"bcr_patient_barcode\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "56f03af0-9904-40ec-ada5-798c23c1f0e1",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BLCA\n",
      "BRCA\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Peri (6-12 months since last menstrual period)_menopause_status have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      ">>> print(df.loc[~events, 'Peri (6-12 months since last menstrual period)_menopause_status'].var())\n",
      "\n",
      "A very low variance means that the column Peri (6-12 months since last menstrual period)_menopause_status completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CESC\n",
      "COAD\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ESCA\n",
      "GBM\n",
      "HNSC\n",
      "KICH\n",
      "KIRC\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Current smoker_tobacco_smoking_history have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Current smoker_tobacco_smoking_history'].var())\n",
      ">>> print(df.loc[~events, 'Current smoker_tobacco_smoking_history'].var())\n",
      "\n",
      "A very low variance means that the column Current smoker_tobacco_smoking_history completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 5:96400118:C:T_T have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '5:96400118:C:T_T'].var())\n",
      ">>> print(df.loc[~events, '5:96400118:C:T_T'].var())\n",
      "\n",
      "A very low variance means that the column 5:96400118:C:T_T completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "KIRP\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column Current smoker_tobacco_smoking_history have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'Current smoker_tobacco_smoking_history'].var())\n",
      ">>> print(df.loc[~events, 'Current smoker_tobacco_smoking_history'].var())\n",
      "\n",
      "A very low variance means that the column Current smoker_tobacco_smoking_history completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LGG\n",
      "LIHC\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 12:69642145:G:T_T have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '12:69642145:G:T_T'].var())\n",
      ">>> print(df.loc[~events, '12:69642145:G:T_T'].var())\n",
      "\n",
      "A very low variance means that the column 12:69642145:G:T_T completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 5:95828707:A:G_G have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '5:95828707:A:G_G'].var())\n",
      ">>> print(df.loc[~events, '5:95828707:A:G_G'].var())\n",
      "\n",
      "A very low variance means that the column 5:95828707:A:G_G completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 5:96357030:G:T_T have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '5:96357030:G:T_T'].var())\n",
      ">>> print(df.loc[~events, '5:96357030:G:T_T'].var())\n",
      "\n",
      "A very low variance means that the column 5:96357030:G:T_T completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 1:89322047:T:C_C have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '1:89322047:T:C_C'].var())\n",
      ">>> print(df.loc[~events, '1:89322047:T:C_C'].var())\n",
      "\n",
      "A very low variance means that the column 1:89322047:T:C_C completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 11:65613446:A:C_C have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '11:65613446:A:C_C'].var())\n",
      ">>> print(df.loc[~events, '11:65613446:A:C_C'].var())\n",
      "\n",
      "A very low variance means that the column 11:65613446:A:C_C completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 5:96326397:T:G_G have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '5:96326397:T:G_G'].var())\n",
      ">>> print(df.loc[~events, '5:96326397:T:G_G'].var())\n",
      "\n",
      "A very low variance means that the column 5:96326397:T:G_G completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 5:96439434:C:G_G have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '5:96439434:C:G_G'].var())\n",
      ">>> print(df.loc[~events, '5:96439434:C:G_G'].var())\n",
      "\n",
      "A very low variance means that the column 5:96439434:C:G_G completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 11:65578204:G:A_A have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '11:65578204:G:A_A'].var())\n",
      ">>> print(df.loc[~events, '11:65578204:G:A_A'].var())\n",
      "\n",
      "A very low variance means that the column 11:65578204:G:A_A completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 16:23748402:G:C_C have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '16:23748402:G:C_C'].var())\n",
      ">>> print(df.loc[~events, '16:23748402:G:C_C'].var())\n",
      "\n",
      "A very low variance means that the column 16:23748402:G:C_C completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 5:95975953:G:A_A have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '5:95975953:G:A_A'].var())\n",
      ">>> print(df.loc[~events, '5:95975953:G:A_A'].var())\n",
      "\n",
      "A very low variance means that the column 5:95975953:G:A_A completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 5:96429646:C:T_T have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '5:96429646:C:T_T'].var())\n",
      ">>> print(df.loc[~events, '5:96429646:C:T_T'].var())\n",
      "\n",
      "A very low variance means that the column 5:96429646:C:T_T completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 11:65110727:G:A_A have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '11:65110727:G:A_A'].var())\n",
      ">>> print(df.loc[~events, '11:65110727:G:A_A'].var())\n",
      "\n",
      "A very low variance means that the column 11:65110727:G:A_A completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 16:23630920:G:T_T have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '16:23630920:G:T_T'].var())\n",
      ">>> print(df.loc[~events, '16:23630920:G:T_T'].var())\n",
      "\n",
      "A very low variance means that the column 16:23630920:G:T_T completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 1:89442053:A:T_T have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '1:89442053:A:T_T'].var())\n",
      ">>> print(df.loc[~events, '1:89442053:A:T_T'].var())\n",
      "\n",
      "A very low variance means that the column 1:89442053:A:T_T completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 5:95897963:A:T_T have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '5:95897963:A:T_T'].var())\n",
      ">>> print(df.loc[~events, '5:95897963:A:T_T'].var())\n",
      "\n",
      "A very low variance means that the column 5:95897963:A:T_T completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 11:64911621:A:G_G have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '11:64911621:A:G_G'].var())\n",
      ">>> print(df.loc[~events, '11:64911621:A:G_G'].var())\n",
      "\n",
      "A very low variance means that the column 11:64911621:A:G_G completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 11:65392357:G:A_A have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '11:65392357:G:A_A'].var())\n",
      ">>> print(df.loc[~events, '11:65392357:G:A_A'].var())\n",
      "\n",
      "A very low variance means that the column 11:65392357:G:A_A completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 5:95982190:T:C_C have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '5:95982190:T:C_C'].var())\n",
      ">>> print(df.loc[~events, '5:95982190:T:C_C'].var())\n",
      "\n",
      "A very low variance means that the column 5:95982190:T:C_C completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 5:96206416:G:T_T have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '5:96206416:G:T_T'].var())\n",
      ">>> print(df.loc[~events, '5:96206416:G:T_T'].var())\n",
      "\n",
      "A very low variance means that the column 5:96206416:G:T_T completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 5:96207276:C:T_T have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '5:96207276:C:T_T'].var())\n",
      ">>> print(df.loc[~events, '5:96207276:C:T_T'].var())\n",
      "\n",
      "A very low variance means that the column 5:96207276:C:T_T completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 5:96218877:G:T_T have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '5:96218877:G:T_T'].var())\n",
      ">>> print(df.loc[~events, '5:96218877:G:T_T'].var())\n",
      "\n",
      "A very low variance means that the column 5:96218877:G:T_T completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 5:96240214:C:A_A have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '5:96240214:C:A_A'].var())\n",
      ">>> print(df.loc[~events, '5:96240214:C:A_A'].var())\n",
      "\n",
      "A very low variance means that the column 5:96240214:C:A_A completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 11:65367356:C:T_T have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '11:65367356:C:T_T'].var())\n",
      ">>> print(df.loc[~events, '11:65367356:C:T_T'].var())\n",
      "\n",
      "A very low variance means that the column 11:65367356:C:T_T completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 5:80100617:C:G_G have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '5:80100617:C:G_G'].var())\n",
      ">>> print(df.loc[~events, '5:80100617:C:G_G'].var())\n",
      "\n",
      "A very low variance means that the column 5:80100617:C:G_G completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 1:88746245:A:G_G have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '1:88746245:A:G_G'].var())\n",
      ">>> print(df.loc[~events, '1:88746245:A:G_G'].var())\n",
      "\n",
      "A very low variance means that the column 1:88746245:A:G_G completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 1:89385281:C:T_T have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '1:89385281:C:T_T'].var())\n",
      ">>> print(df.loc[~events, '1:89385281:C:T_T'].var())\n",
      "\n",
      "A very low variance means that the column 1:89385281:C:T_T completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 5:96324943:G:T_T have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '5:96324943:G:T_T'].var())\n",
      ">>> print(df.loc[~events, '5:96324943:G:T_T'].var())\n",
      "\n",
      "A very low variance means that the column 5:96324943:G:T_T completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 7:80290369:G:A_A have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '7:80290369:G:A_A'].var())\n",
      ">>> print(df.loc[~events, '7:80290369:G:A_A'].var())\n",
      "\n",
      "A very low variance means that the column 7:80290369:G:A_A completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 1:89119930:T:C_C have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '1:89119930:T:C_C'].var())\n",
      ">>> print(df.loc[~events, '1:89119930:T:C_C'].var())\n",
      "\n",
      "A very low variance means that the column 1:89119930:T:C_C completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 1:89442977:C:G_G have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '1:89442977:C:G_G'].var())\n",
      ">>> print(df.loc[~events, '1:89442977:C:G_G'].var())\n",
      "\n",
      "A very low variance means that the column 1:89442977:C:G_G completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 5:96429750:G:A_A have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '5:96429750:G:A_A'].var())\n",
      ">>> print(df.loc[~events, '5:96429750:G:A_A'].var())\n",
      "\n",
      "A very low variance means that the column 5:96429750:G:A_A completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 12:113403675:C:T_T have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '12:113403675:C:T_T'].var())\n",
      ">>> print(df.loc[~events, '12:113403675:C:T_T'].var())\n",
      "\n",
      "A very low variance means that the column 12:113403675:C:T_T completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 5:96447637:G:A_A have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '5:96447637:G:A_A'].var())\n",
      ">>> print(df.loc[~events, '5:96447637:G:A_A'].var())\n",
      "\n",
      "A very low variance means that the column 5:96447637:G:A_A completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 5:80055860:C:T_T have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '5:80055860:C:T_T'].var())\n",
      ">>> print(df.loc[~events, '5:80055860:C:T_T'].var())\n",
      "\n",
      "A very low variance means that the column 5:80055860:C:T_T completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 5:95967581:G:A_A have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '5:95967581:G:A_A'].var())\n",
      ">>> print(df.loc[~events, '5:95967581:G:A_A'].var())\n",
      "\n",
      "A very low variance means that the column 5:95967581:G:A_A completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 12:69766606:C:T_T have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '12:69766606:C:T_T'].var())\n",
      ">>> print(df.loc[~events, '12:69766606:C:T_T'].var())\n",
      "\n",
      "A very low variance means that the column 12:69766606:C:T_T completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 3:38390884:C:T_T have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '3:38390884:C:T_T'].var())\n",
      ">>> print(df.loc[~events, '3:38390884:C:T_T'].var())\n",
      "\n",
      "A very low variance means that the column 3:38390884:C:T_T completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 5:79683830:T:C_C have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '5:79683830:T:C_C'].var())\n",
      ">>> print(df.loc[~events, '5:79683830:T:C_C'].var())\n",
      "\n",
      "A very low variance means that the column 5:79683830:T:C_C completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 7:136075342:C:T_T have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '7:136075342:C:T_T'].var())\n",
      ">>> print(df.loc[~events, '7:136075342:C:T_T'].var())\n",
      "\n",
      "A very low variance means that the column 7:136075342:C:T_T completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 5:96218022:G:T_T have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '5:96218022:G:T_T'].var())\n",
      ">>> print(df.loc[~events, '5:96218022:G:T_T'].var())\n",
      "\n",
      "A very low variance means that the column 5:96218022:G:T_T completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column G4_neoplasm_histologic_grade have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, 'G4_neoplasm_histologic_grade'].var())\n",
      ">>> print(df.loc[~events, 'G4_neoplasm_histologic_grade'].var())\n",
      "\n",
      "A very low variance means that the column G4_neoplasm_histologic_grade completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LUAD\n",
      "LUSC\n",
      "OV\n",
      "PAAD\n",
      "PCPG\n",
      "PRAD\n",
      "READ\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 5:95828707:A:G_G have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '5:95828707:A:G_G'].var())\n",
      ">>> print(df.loc[~events, '5:95828707:A:G_G'].var())\n",
      "\n",
      "A very low variance means that the column 5:95828707:A:G_G completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 16:23712903:G:A_A have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '16:23712903:G:A_A'].var())\n",
      ">>> print(df.loc[~events, '16:23712903:G:A_A'].var())\n",
      "\n",
      "A very low variance means that the column 16:23712903:G:A_A completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 3:133637892:T:C_C have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '3:133637892:T:C_C'].var())\n",
      ">>> print(df.loc[~events, '3:133637892:T:C_C'].var())\n",
      "\n",
      "A very low variance means that the column 3:133637892:T:C_C completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 1:89312813:G:A_A have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '1:89312813:G:A_A'].var())\n",
      ">>> print(df.loc[~events, '1:89312813:G:A_A'].var())\n",
      "\n",
      "A very low variance means that the column 1:89312813:G:A_A completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SARC\n",
      "SKCM\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 7:136075342:C:T_T have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '7:136075342:C:T_T'].var())\n",
      ">>> print(df.loc[~events, '7:136075342:C:T_T'].var())\n",
      "\n",
      "A very low variance means that the column 7:136075342:C:T_T completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "STAD\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 1:161463042:A:G_G have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '1:161463042:A:G_G'].var())\n",
      ">>> print(df.loc[~events, '1:161463042:A:G_G'].var())\n",
      "\n",
      "A very low variance means that the column 1:161463042:A:G_G completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TGCT\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 5:95982245:G:T_T have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '5:95982245:G:T_T'].var())\n",
      ">>> print(df.loc[~events, '5:95982245:G:T_T'].var())\n",
      "\n",
      "A very low variance means that the column 5:95982245:G:T_T completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 5:96231200:C:T_T have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '5:96231200:C:T_T'].var())\n",
      ">>> print(df.loc[~events, '5:96231200:C:T_T'].var())\n",
      "\n",
      "A very low variance means that the column 5:96231200:C:T_T completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 5:96324943:G:T_T have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '5:96324943:G:T_T'].var())\n",
      ">>> print(df.loc[~events, '5:96324943:G:T_T'].var())\n",
      "\n",
      "A very low variance means that the column 5:96324943:G:T_T completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 11:65407886:C:T_T have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '11:65407886:C:T_T'].var())\n",
      ">>> print(df.loc[~events, '11:65407886:C:T_T'].var())\n",
      "\n",
      "A very low variance means that the column 11:65407886:C:T_T completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/utils/__init__.py:1122: ConvergenceWarning: Column 11:65650883:A:G_G have very low variance when conditioned on death event present or not. This may harm convergence. This could be a form of 'complete separation'. For example, try the following code:\n",
      "\n",
      ">>> events = df['PFI'].astype(bool)\n",
      ">>> print(df.loc[events, '11:65650883:A:G_G'].var())\n",
      ">>> print(df.loc[~events, '11:65650883:A:G_G'].var())\n",
      "\n",
      "A very low variance means that the column 11:65650883:A:G_G completely determines whether a subject dies or not. See https://stats.stackexchange.com/questions/11109/how-to-deal-with-perfect-separation-in-logistic-regression.\n",
      "\n",
      "  warnings.warn(dedent(warning_text), ConvergenceWarning)\n",
      "/cellar/users/tsears/.local/lib/python3.7/site-packages/lifelines/fitters/coxph_fitter.py:1612: ConvergenceWarning: Newton-Rhaphson failed to converge sufficiently. Please see the following tips in the lifelines documentation: https://lifelines.readthedocs.io/en/latest/Examples.html#problems-with-convergence-in-the-cox-proportional-hazard-model\n",
      "  \"Newton-Rhaphson failed to converge sufficiently. {0}\".format(CONVERGENCE_DOCS), exceptions.ConvergenceWarning\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "THCA\n",
      "THYM\n",
      "UCEC\n"
     ]
    }
   ],
   "source": [
    "#run coxph analysis on test datasets, keep cov column for analysis on training datasets\n",
    "\n",
    "df_coxph = pd.DataFrame()\n",
    "cancer_test=[]\n",
    "cov_cols=[]\n",
    "\n",
    "for i,row in supp_table[supp_table[\"index\"].isin(canc_keep)].iterrows():\n",
    "    print(row[\"index\"])\n",
    "    \n",
    "    for snp in rsid_lst:\n",
    "        temp_df = os_test[os_test[\"cancer\"]==row[\"index\"]]\n",
    "        temp_df=temp_df[~temp_df[snp].isnull()]\n",
    "        \n",
    "        #if max(temp_df[snp])==0:\n",
    "        #    continue\n",
    "        \n",
    "        if len(temp_df[snp].value_counts())>1:       \n",
    "            cov_col=row[\"var\"].split(\",\")\n",
    "            for x in cov_col:\n",
    "                if x in [x for x in mp_dummy.keys()]:\n",
    "                    cov_col=cov_col+mp_dummy[x]\n",
    "            cov_col=[x for x in cov_col if x in surv.columns]\n",
    "            cov_col=[x for x in cov_col if len(temp_df[x].value_counts()) > 1]\n",
    "            \n",
    "            if len(temp_df[\"gender\"].value_counts())>1:\n",
    "                temp_df = temp_df[['age_at_initial_pathologic_diagnosis',snp,\"gender\",\"PFI\",'PFI.time']+cov_col]\n",
    "            else:\n",
    "                temp_df = temp_df[['age_at_initial_pathologic_diagnosis',snp,\"PFI\",'PFI.time']+cov_col]\n",
    "                \n",
    "            #possibly use this line?\n",
    "            #temp_df=temp_df.drop(list(temp_df.var().index[temp_df.var()<0.01]),axis=1)\n",
    "            \n",
    "            temp_df=temp_df.fillna(0)\n",
    "            cph = CoxPHFitter(penalizer=0.01)\n",
    "            cph.fit(temp_df, event_col=\"PFI\", duration_col='PFI.time')\n",
    "            cancer_test.append(row[\"index\"])\n",
    "            cov_cols.append(cov_col)\n",
    "            df_coxph  = df_coxph.append(cph.summary.loc[[snp]])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "ea8de23e-87c1-4ebb-b092-1c0148a851a2",
   "metadata": {},
   "outputs": [],
   "source": [
    "x=pd.DataFrame(zip(cancer_test,cov_cols))\n",
    "cov_col_table=x[~x[0].duplicated()]\n",
    "cov_col_table=cov_col_table.set_index(0)\n",
    "cov_col_table[\"index\"]=cov_col_table.index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "aa55d40a-a638-47ee-9c1a-fbb009e6138d",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BLCA\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "/cm/shared/apps/jupyter/12.3.0/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n"
     ]
    },
    {
     "ename": "KeyError",
     "evalue": "\"None of [Index(['11:65613446:A:C_C'], dtype='object', name='covariate')] are in the [index]\"",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m/tmp/ipykernel_2495750/1438445032.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     26\u001b[0m         \u001b[0mcph\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtemp_df\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mevent_col\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"PFI\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mduration_col\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'PFI.time'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     27\u001b[0m         \u001b[0mcancer_train\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrow\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"index\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 28\u001b[0;31m         \u001b[0mdf_coxph_train\u001b[0m  \u001b[0;34m=\u001b[0m \u001b[0mdf_coxph_train\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcph\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msummary\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0msnp\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/.local/lib/python3.7/site-packages/pandas/core/indexing.py\u001b[0m in \u001b[0;36m__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m    929\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    930\u001b[0m             \u001b[0mmaybe_callable\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcom\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply_if_callable\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mobj\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 931\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_getitem_axis\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmaybe_callable\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0maxis\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    932\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    933\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_is_scalar_access\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mtuple\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.7/site-packages/pandas/core/indexing.py\u001b[0m in \u001b[0;36m_getitem_axis\u001b[0;34m(self, key, axis)\u001b[0m\n\u001b[1;32m   1151\u001b[0m                     \u001b[0;32mraise\u001b[0m \u001b[0mValueError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"Cannot index with multidimensional key\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1152\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1153\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_getitem_iterable\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0maxis\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1154\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1155\u001b[0m             \u001b[0;31m# nested tuple slicing\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.7/site-packages/pandas/core/indexing.py\u001b[0m in \u001b[0;36m_getitem_iterable\u001b[0;34m(self, key, axis)\u001b[0m\n\u001b[1;32m   1091\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1092\u001b[0m         \u001b[0;31m# A collection of keys\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1093\u001b[0;31m         \u001b[0mkeyarr\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mindexer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_listlike_indexer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1094\u001b[0m         return self.obj._reindex_with_indexers(\n\u001b[1;32m   1095\u001b[0m             \u001b[0;34m{\u001b[0m\u001b[0maxis\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0mkeyarr\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mindexer\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcopy\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mallow_dups\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.7/site-packages/pandas/core/indexing.py\u001b[0m in \u001b[0;36m_get_listlike_indexer\u001b[0;34m(self, key, axis)\u001b[0m\n\u001b[1;32m   1312\u001b[0m             \u001b[0mkeyarr\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mindexer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnew_indexer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0max\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_reindex_non_unique\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkeyarr\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1313\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1314\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_validate_read_indexer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkeyarr\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mindexer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1315\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1316\u001b[0m         if needs_i8_conversion(ax.dtype) or isinstance(\n",
      "\u001b[0;32m~/.local/lib/python3.7/site-packages/pandas/core/indexing.py\u001b[0m in \u001b[0;36m_validate_read_indexer\u001b[0;34m(self, key, indexer, axis)\u001b[0m\n\u001b[1;32m   1372\u001b[0m                 \u001b[0;32mif\u001b[0m \u001b[0muse_interval_msg\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1373\u001b[0m                     \u001b[0mkey\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlist\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1374\u001b[0;31m                 \u001b[0;32mraise\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34mf\"None of [{key}] are in the [{axis_name}]\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1375\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1376\u001b[0m             \u001b[0mnot_found\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlist\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mensure_index\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mmissing_mask\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mnonzero\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0munique\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: \"None of [Index(['11:65613446:A:C_C'], dtype='object', name='covariate')] are in the [index]\""
     ]
    }
   ],
   "source": [
    "#run coxph analysis on TRAIN datasets, use cov column from analysis on test datasets\n",
    "\n",
    "df_coxph_train = pd.DataFrame()\n",
    "cancer_train=[]\n",
    "\n",
    "for i,row in cov_col_table.iterrows():\n",
    "    print(row[\"index\"])\n",
    "    \n",
    "    for snp in rsid_lst:\n",
    "        temp_df = os_train[os_train[\"cancer\"]==row[\"index\"]]\n",
    "        temp_df=temp_df[~temp_df[snp].isnull()]\n",
    "        \n",
    "        cov_col=row[1]\n",
    "            \n",
    "        if max(temp_df[snp])==0:\n",
    "            continue\n",
    "        if len(temp_df[\"gender\"].value_counts())>1:\n",
    "            temp_df = temp_df[['age_at_initial_pathologic_diagnosis',snp,\"gender\",\"PFI\",'PFI.time']+cov_col]\n",
    "        else:\n",
    "            temp_df = temp_df[['age_at_initial_pathologic_diagnosis',snp,\"PFI\",'PFI.time']+cov_col] \n",
    "        \n",
    "        temp_df=temp_df.drop(list(temp_df.var().index[temp_df.var()<0.01]),axis=1) #seems to be necessary\n",
    "\n",
    "        temp_df=temp_df.fillna(0)\n",
    "        cph = CoxPHFitter(penalizer=0.01)\n",
    "        cph.fit(temp_df, event_col=\"PFI\", duration_col='PFI.time')\n",
    "        cancer_train.append(row[\"index\"])\n",
    "        df_coxph_train  = df_coxph_train.append(cph.summary.loc[[snp]])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b0620fd3-43c7-4182-832b-bbb5be234e86",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Insert snp character column, add cancer index\n",
    "\n",
    "df_coxph[\"proxy_variant_chars\"]=df_coxph.index\n",
    "df_coxph_train[\"proxy_variant_chars\"]=df_coxph_train.index\n",
    "\n",
    "df_coxph[\"cancer\"]=cancer_test\n",
    "df_coxph_train[\"cancer\"]=cancer_train"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1bb45eba-9676-436f-bb44-e9e53ef19089",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Apply beta values from respective coxph dataset to respective surv dataset\n",
    "os_test_beta=os_test.copy()\n",
    "os_train_beta=os_train.copy()\n",
    "\n",
    "for i,row in df_coxph.iterrows():\n",
    "    if row[\"proxy_variant_chars\"] in os_test and not np.isnan(row[\"coef\"]):\n",
    "        os_test_beta.loc[os_test_beta[\"cancer\"]==row[\"cancer\"],i]=os_test.loc[os_test[\"cancer\"]==row[\"cancer\"],i]*row[\"coef\"]\n",
    "\n",
    "for i,row in df_coxph_train.iterrows():\n",
    "    if row[\"proxy_variant_chars\"] in os_train and not np.isnan(row[\"coef\"]):\n",
    "        os_train_beta.loc[os_train_beta[\"cancer\"]==row[\"cancer\"],i]=os_train[i][os_train[\"cancer\"]==row[\"cancer\"]]*row[\"coef\"]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c81fa848-a110-40fa-add4-85c0f65695e6",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Save beta dataframes\n",
    "#os_train_beta.to_csv(\"/cellar/users/tsears/projects/germline-immune/data/fig5/PFI_train_beta.csv\",sep=\"\\t\")\n",
    "#os_test_beta.to_csv(\"/cellar/users/tsears/projects/germline-immune/data/fig5/PFI_test_beta.csv\",sep=\"\\t\")\n",
    "#os_train_beta=pd.read_csv(\"/cellar/users/tsears/projects/germline-immune/data/fig5/PFI_train_beta.csv\",sep=\"\\t\")\n",
    "#os_test_beta=pd.read_csv(\"/cellar/users/tsears/projects/germline-immune/data/fig5/PFI_test_beta.csv\",sep=\"\\t\")\n",
    "#cov_col_table.to_csv(\"/cellar/users/tsears/projects/germline-immune/data/fig5/PFS_covariate_table\",sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "ca470603-3f14-4155-a772-2036c5ad5e96",
   "metadata": {},
   "outputs": [],
   "source": [
    "#cov_col_table.loc[\"LUAD\"][1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "97d54063-ac9d-4b73-b0fa-5fc01c2fb2d5",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(296, 135)\n",
      "0.542695035460993\n",
      "SNPs dropped =  30 out of  37\n",
      "mannWhitnetPredScore= 0.15527448594485227\n",
      "LUAD\n",
      "coef                   0.035390\n",
      "exp(coef)              1.036023\n",
      "se(coef)               0.015954\n",
      "coef lower 95%         0.004120\n",
      "coef upper 95%         0.066659\n",
      "exp(coef) lower 95%    1.004129\n",
      "exp(coef) upper 95%    1.068931\n",
      "cmp to                 0.000000\n",
      "z                      2.218236\n",
      "p                      0.026539\n",
      "-log2(p)               5.235757\n",
      "Name: LASSO_burden, dtype: float64\n"
     ]
    }
   ],
   "source": [
    "#Create grand for loop to produce plots for everything\n",
    "from sklearn.linear_model import Lasso\n",
    "import sklearn.metrics as metrics\n",
    "import xgboost\n",
    "from sklearn.model_selection import cross_val_score\n",
    "from sklearn.model_selection import RepeatedKFold\n",
    "    \n",
    "#Loop through lasso, drop snps, input list of hyperparams as well as respective cancer types\n",
    "#store coxph outcomes into final df for easy final plotting\n",
    "#run for loop by index and use 3 separate lists for input info?\n",
    "\n",
    "#### part 1, lasso model ####\n",
    "cancer_list=cov_col_table[\"index\"]\n",
    "cancer_list=cancer_list.drop(\"TGCT\")\n",
    "cancer_list=cancer_list.drop(\"THCA\")\n",
    "km_curve=\"on\"\n",
    "\n",
    "\n",
    "alphas=[0.0001,0.005,0.01,0.000005,0.001,0.0001,0.0001,0.01,0.00001,0.009,0.0001,0.0001,0.00001,0.00001,0.001]\n",
    "#.      BLCA.   BRCA. COAD     ESCA.   HNSC  KICH  KIRC.  KIRP.  LIHC  LUAD.   LUSC.  PAAD  READ. SKCM    STAD.   \n",
    "\n",
    "\n",
    "hazard_list=pd.DataFrame()\n",
    "all_survival_snps=pd.DataFrame()\n",
    "\n",
    "\n",
    "for i in range(len(cancer_list)):\n",
    "    \n",
    "    #Filter down dataframes to snps used in respective cancer type\n",
    "    surv_fdr_drop=surv_fdr[surv_fdr[\"cancer\"]==cancer_list[i]]\n",
    "    surv_fdr_drop=surv_fdr_drop[surv_fdr_drop[\"survival\"]==\"PFI\"]\n",
    "    surv_fdr_drop=surv_fdr_drop[surv_fdr_drop[\"logrank p\"]<0.05]\n",
    "    \n",
    "    cols_to_drop=np.setdiff1d(os_train_beta.filter(like=\":\").columns.values.tolist(),surv_fdr_drop[\"snps\"].to_list()).tolist()\n",
    "    \n",
    "    train_df=os_train_beta[os_train_beta[\"cancer\"]==cancer_list[i]]\n",
    "    test_df=os_test_beta[os_test_beta[\"cancer\"]==cancer_list[i]]\n",
    "    \n",
    "    train_df=train_df.drop(cols_to_drop,axis=1)\n",
    "    test_df=test_df.drop(cols_to_drop,axis=1)\n",
    "    \n",
    "    print(train_df.shape)\n",
    "    \n",
    "    train_y=train_df[\"PFI\"]\n",
    "    test_y=test_df[\"PFI\"]\n",
    "    \n",
    "    train_x=train_df.filter(like=\":\")\n",
    "    test_x=test_df.filter(like=\":\")\n",
    "\n",
    "    #pd.Series(train_x.columns.values).to_csv(\"/cellar/users/tsears/projects/germline-immune/data/fig5/\"+cancer_list[i]+\"_filtered_SNP_list_PFI.txt\",sep=\"\\t\",index=False,header=False)\n",
    " \n",
    "    #filter using LASSO regression\n",
    "\n",
    "    # define mode\n",
    "    model = Lasso(alpha=alphas[i],random_state=random_state)\n",
    "    # fit model\n",
    "    model.fit(train_x, train_y)\n",
    "    p = model.predict(test_x)  # test the output by changing values\n",
    "\n",
    "    #Find model that gives best AUC, do a cutoff of weakest contributing SNPS, then run all analysis portions\n",
    "    # calculate the fpr and tpr for all thresholds of the classification\n",
    "    #preds = np.around(p, decimals=0, out=None)\n",
    "\n",
    "    preds=p\n",
    "\n",
    "    fpr, tpr, threshold = metrics.roc_curve(list(test_y), preds,pos_label=1)\n",
    "    roc_auc = metrics.auc(fpr, tpr)\n",
    "\n",
    "    print(roc_auc)\n",
    "    import matplotlib.pyplot as plt\n",
    "\n",
    "    #### part 2 ####\n",
    "\n",
    "    #remake SNP list with dropped \"useless\" SNPs\n",
    "    \n",
    "    print(\"SNPs dropped = \",len(train_x.columns.values[model.coef_==0].tolist()),\"out of \",len(train_x.columns.values))\n",
    "    train_df=train_df.drop(train_x.columns.values[model.coef_==0].tolist(),axis=1)\n",
    "    test_df=test_df.drop(test_x.columns.values[model.coef_==0].tolist(),axis=1)\n",
    "    \n",
    "    surv_fdr_drop=surv_fdr_drop[~surv_fdr_drop[\"snps\"].isin(test_x.columns.values[model.coef_==0].tolist())]\n",
    "    all_survival_snps=all_survival_snps.append(surv_fdr_drop)\n",
    "    \n",
    "    train_x=train_df.filter(like=\":\")\n",
    "    test_x=test_df.filter(like=\":\")\n",
    "    \n",
    "    regressor_xgb = xgboost.XGBRFRegressor(n_estimators = 300, random_state = random_state,learning_rate=0.5,max_depth=9)\n",
    "    regressor_xgb.fit(train_x, train_y)  \n",
    "\n",
    "    p = regressor_xgb.predict(train_x)  # test the output by changing values\n",
    "\n",
    "    test_df[\"LASSO_burden\"]=p*100\n",
    "    train_df[\"LASSO_burden\"]=p_disc*100\n",
    "\n",
    "    #fig = plt.figure(figsize=(5,5))\n",
    "\n",
    "    #sns.violinplot(x=\"OS\",y=\"LASSO_burden\",data=test_df)\n",
    "\n",
    "    cat1 = test_df[test_df[\"PFI\"]==0]\n",
    "    cat2 = test_df[test_df[\"PFI\"]==1]\n",
    "    print(\"mannWhitnetPredScore=\",stats.mannwhitneyu(cat1[\"LASSO_burden\"], cat2[\"LASSO_burden\"])[1])\n",
    "\n",
    "    #plt.title(stats.mannwhitneyu(cat1[\"LASSO_burden\"], cat2[\"LASSO_burden\"])[1])\n",
    "\n",
    "    #plt.ylabel(\"burden (pvalue < 0.05) RF Regression\")\n",
    "    #plt.xticks(rotation=90)\n",
    "    #plt.savefig(\"/cellar/users/tsears/projects/germline-immune/plots/fig5/\"+cancer_list[i]+\"violinplot.pdf\")\n",
    "\n",
    "    #now do hazard plot on PRS scores generated...\n",
    "    test_df_coxPH=test_df[list(np.append(['Stage II', 'Stage III', 'Stage I','ajcc_pathologic_tumor_stage',\n",
    "            'PFI', 'PFI.time',\n",
    "           'age_at_initial_pathologic_diagnosis',\n",
    "           'LASSO_burden'],cov_col_table.loc[cancer_list[i],1]))]\n",
    "    \n",
    "    #filter covariates that throw errors due to low variance\n",
    "    test_df_coxPH=test_df_coxPH.drop(list(test_df_coxPH.var().index[test_df_coxPH.var()<0.01]),axis=1) #seems to be necessary\n",
    "    test_df_coxPH=test_df_coxPH.fillna(0)\n",
    "\n",
    "    cph = CoxPHFitter(penalizer=0.01)\n",
    "\n",
    "    cph.fit(test_df_coxPH, duration_col = 'PFI.time', event_col = 'PFI')\n",
    "    print(cancer_list[i])\n",
    "    #cph.print_summary()\n",
    "    print(cph.summary.loc[\"LASSO_burden\"])\n",
    "    \n",
    "    hazard_list=hazard_list.append(cph.summary.loc[\"LASSO_burden\"])\n",
    "    \n",
    "    if km_curve==\"on\":\n",
    "        #plot KM curve\n",
    "        surv_df_canc=test_df_coxPH\n",
    "\n",
    "        fig=plt.figure(figsize=(5,5))\n",
    "        ax1 = plt.subplot(1,1,1)\n",
    "\n",
    "        quantiles=np.quantile(surv_df_canc[\"LASSO_burden\"],[0,0.25,0.75,1])\n",
    "        surv_df_canc['group'] = pd.cut(surv_df_canc[\"LASSO_burden\"], quantiles, labels=[0,1,2])\n",
    "\n",
    "        kmf = KaplanMeierFitter()\n",
    "        kmf.fit(surv_df_canc[surv_df_canc['group']==2]['PFI.time'], surv_df_canc[surv_df_canc['group']==2]['PFI'],label=\"High, N=\"+str(surv_df_canc[surv_df_canc['group']==2].shape[0]))\n",
    "        kmf.plot(ci_show=False, ax=ax1)\n",
    "        kmf.fit(surv_df_canc[surv_df_canc['group']==1]['PFI.time'], surv_df_canc[surv_df_canc['group']==1]['PFI'],label=\"Med, N=\"+str(surv_df_canc[surv_df_canc['group']==1].shape[0]))\n",
    "        kmf.plot(ci_show=False, ax=ax1)\n",
    "        kmf.fit(surv_df_canc[surv_df_canc['group']==0]['PFI.time'], surv_df_canc[surv_df_canc['group']==0]['PFI'],label=\"Low, N=\"+str(surv_df_canc[surv_df_canc['group']==0].shape[0]))\n",
    "        kmf.plot(ci_show=False, ax=ax1)\n",
    "\n",
    "        results = logrank_test(surv_df_canc[surv_df_canc[\"group\"]==2]['PFI.time'], surv_df_canc[surv_df_canc[\"group\"]==0]['PFI.time'],event_observed_A=surv_df_canc[surv_df_canc[\"group\"]==2]['PFI'], event_observed_B=surv_df_canc[surv_df_canc[\"group\"]==0]['PFI'], alpha=.95)\n",
    "        results2 = logrank_test(surv_df_canc[surv_df_canc[\"group\"]==2]['PFI.time'], surv_df_canc[surv_df_canc[\"group\"]==1]['PFI.time'],event_observed_A=surv_df_canc[surv_df_canc[\"group\"]==2]['PFI'], event_observed_B=surv_df_canc[surv_df_canc[\"group\"]==1]['PFI'], alpha=.95)\n",
    "        results3 = logrank_test(surv_df_canc[surv_df_canc[\"group\"]==1]['PFI.time'], surv_df_canc[surv_df_canc[\"group\"]==0]['PFI.time'],event_observed_A=surv_df_canc[surv_df_canc[\"group\"]==1]['PFI'], event_observed_B=surv_df_canc[surv_df_canc[\"group\"]==0]['PFI'], alpha=.95)\n",
    "\n",
    "        plt.ylabel('% survival')\n",
    "        plt.xlabel('PFI time (days)')\n",
    "        plt.title(str(cancer_list[i])+\"\\n\"+\"High vs Low \"+str(np.round(results.p_value,4))+\"\\n\"+\"High vs Med \"+str(np.round(results2.p_value,4))+\"\\n\"+\"Med vs Low \"+str(np.round(results3.p_value,4)))\n",
    "        plt.legend(frameon=False)\n",
    "\n",
    "        plt.savefig(\"/cellar/users/tsears/projects/germline-immune/plots/fig5/\"+cancer_list[i]+\"_0.25_0.75_PFI_KM.pdf\")\n",
    "        \n",
    "#all_survival_snps.to_csv(\"/cellar/users/tsears/projects/germline-immune/data/fig6/PFS_input_snps.txt\",sep=\"\\t\")\n",
    "\n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "c1fb3342-55e3-4e65-9c44-ebb31080db39",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<matplotlib.lines.Line2D at 0x1554fb337290>"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA34AAADQCAYAAABY17MXAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjQuMywgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/MnkTPAAAACXBIWXMAAAsTAAALEwEAmpwYAAApX0lEQVR4nO3dfbhcZX3v//dHSDEqECCBaABRy4MG5cHE5yo+glULaIVSWrXWg+f8mlp/p8FKUxqRw08s9HfOsWkL1KZgL4tSCxQtB3ygiC1YEyBAgoIoSojEADGCJcU0fM8fszZMdvbDbNiT2Xvm/bquufbMfa+Z/b1nrVlrvnPf616pKiRJkiRJ/etpvQ5AkiRJktRdJn6SJEmS1OdM/CRJkiSpz5n4SZIkSVKfM/GTJEmSpD5n4idJkiRJfW7nXgcwmWbPnl0HHHBAr8OQJEmSpJ648cYbH6iqOcPL+yrxO+CAA1i5cmWvw5AkSZKknkjyw5HKuzbUM8nyJBuSrB6l/tgktyZZlWRlkte01W1tylcluaJbMUqSJEnSIOhmj9+FwDLgM6PUfw24oqoqyUuAS4BDmrrNVXV4F2OTJEmSpIHRtR6/qroO2DhG/c+qqpqHzwRqtGUlSZIkSU9eT2f1THJ8ku8A/wS8v63q6c3wz28mOW6c1zilWXbl/fff381wJUmSJGla6mniV1WXVdUhwHHAmW1Vz62qBcCvA/8ryQvGeI0LqmpBVS2YM2e7yWskSZIkaeBNiev4NcNCn59kdvN4XfP3+8C1wBG9i06SJEmSnrDmoCNZc9CRvQ5jQnqW+CX5xSRp7h8J7AI8mGSPJLs05bOBVwO39ypOSZIkSZruujarZ5KLgaOA2UnuBZYCMwCq6jzgXcB7kmwBNgMnNjN8vhA4P8ljtBLTs6vKxE+SJEmSnqSuJX5VddI49Z8EPjlC+fXAi7sVlyRJkiQNmilxjp8kSZIkqXtM/CRJkiSpz5n4SZIkSVKfM/GTJEmSpD5n4idJkiRJfc7ET5IkSZL6nImfJEmSJPU5Ez9JkiRJ6nMmfpIkSZLU50z8JEmSJKnPmfhJkiRJUp/rauKXZHmSDUlWj1J/bJJbk6xKsjLJa9rq3pvku83tvd2MU5IkSZL6Wbd7/C4Ejhmj/mvAYVV1OPB+4NMASfYElgIvB14GLE2yR1cjlSRJkqQ+1dXEr6quAzaOUf+zqqrm4TOBoftHA1+pqo1V9RPgK4ydQEqSJEmSRtHzc/ySHJ/kO8A/0er1A5gHrG1b7N6mTJIkSZI0QT1P/Krqsqo6BDgOOHOiz09ySnN+4Mr7779/0uOTJEmSpOmu54nfkGZY6POTzAbWAfu1Ve/blI30vAuqakFVLZgzZ84OiFSSJEmSppeeJn5JfjFJmvtHArsADwJXA29JskczqctbmjJJkiRJ0gTt3M0XT3IxcBQwO8m9tGbqnAFQVecB7wLek2QLsBk4sZnsZWOSM4EVzUt9vKpGnSRGkiRJkjS6riZ+VXXSOPWfBD45St1yYHk34pIkSZKkQTJlzvGTJEmSJHWHiZ8kSZIk9TkTP0mSJEnqcyZ+kiRJktTnTPwkSZIkqc+Z+EmSJElSnzPxkyRJkqQ+Z+InSZIkSR1acdYy9lx/Dy/87irW77EPK85a1uuQOtLVC7hLkiRJUr9YcdYyDj1jMTO3PArA3E0b2P2MxawAFi5Z1NvgxmGPnyRJkiR1YL9zz3w86Rsyc8uj7HfumT2KqHMmfpIkSZLUgb033T+h8qnExE+SJEmSOrBh1pwJlU8lXUv8kixPsiHJ6lHqT05ya5Lbklyf5LC2uh805auSrOxWjJIkSZLUqbWLT2fzjF22Kds8YxfWLj69RxF1rps9fhcCx4xRfzfwuqp6MXAmcMGw+tdX1eFVtaBL8UmSJElSxxYuWcTqpedy36578Rhh/ay9Wb303Ck/sQt0cVbPqrouyQFj1F/f9vCbwL7dikWSJEmSJsPCJYtYc9FyNs7dn/l33sTcXgfUoalyjt9vA/+n7XEBX05yY5JTxnpiklOSrEyy8v77p/5JlZIkSZK0o/X8On5JXk8r8XtNW/Frqmpdkr2BryT5TlVdN9Lzq+oCmmGiCxYsqK4HLEmSJEnTTE97/JK8BPg0cGxVPThUXlXrmr8bgMuAl/UmQkmSJEma/nqW+CXZH7gU+M2qurOt/JlJdh26D7wFGHFmUEmSJEnS+Lo21DPJxcBRwOwk9wJLgRkAVXUe8MfAXsBfJAH4z2YGz32Ay5qynYG/q6qruhWnJEmSJPW7jhK/JPsAC5uH32qGYI6pqk4ap/4DwAdGKP8+cNj2z5AkSZKmlxPPvwGAz3/wlT2ORINu3KGeSU4AvgW8GzgB+Lckv9rtwCRJkiRJk6OTHr8lwMKhXr4kc4CvAl/oZmCSJEmSpMnRyeQuTxs2tPPBDp8nSZIkSZoCOunxuyrJ1cDFzeMTgSu7F5IkSZIkaTKNm/hV1alJ3gW8uim6oKou625YkiRJkqTJ0tGsnlX1D8A/dDkWSZIkSVIXjJr4JfmXqnpNkoeBaq8Cqqp263p0kiRJkqSnbNTEr6pe0/zddceFI0mSJEmabJ1cx+9vOymTJEmSJE1NnVyWYX77gyQ7Ay/tTjiSJEmSpMk2auKX5LTm/L6XJHmouT0M/Bj4xx0WoSRJkiTpKRk18auqTzTn951TVbs1t12raq+qOm28F06yPMmGJKtHqT85ya1JbktyfZLD2uqOSXJHkruSfPRJtUySJEnSDrPmoCNZc9CRvQ5Do+jkOn6nJdkDOBB4elv5deM89UJgGfCZUervBl5XVT9J8lbgAuDlSXYC/hx4M3AvsCLJFVV1+3ixSpIkSZK2N27il+QDwO8B+wKrgFcANwBvGOt5VXVdkgPGqL++7eE3m9cHeBlwV1V9v/n/nwOOBUz8JEmS+sRQz9D8O2/qcSTSYOhkcpffAxYCP6yq1wNHAJsmOY7fBv5Pc38esLat7t6mTJIkSZL0JIzb4wf8R1X9RxKS7FJV30ly8GQFkOT1tBK/1zzJ558CnAKw//77T1ZYkiRJktQ3OunxuzfJLOBy4CtJ/hH44WT88yQvAT4NHFtVDzbF64D92hbbtykbUVVdUFULqmrBnDlzJiMsSZIkSeornUzucnxz92NJ/hnYnSeGZT5pSfYHLgV+s6rubKtaARyY5Hm0Er5fA379qf4/SZIkSRpUnQz1fFxVfT3JW4Arac26OaokFwNHAbOT3AssBWY0r3Me8MfAXsBfJAH4z6bn7j+TLAKuBnYCllfVmgm1SpIkSZL0uFETvyRvAM4DnkNrmOcngb8BApw13gtX1Unj1H8A+MAodVfSSi4lSZIkSU/RWOf4/SmtSVP2Ar5A6xIOF1bVS6vq0h0RnCRJkiTpqRtrqGdV1bXN/cuTrKuqZTsgJkmSJEnSJBor8ZuV5J3ty7Y/ttdPkiRJkqaHsRK/rwPvaHt8XdvjojUjpyRJkiRpihs18auq39qRgUiSJA26NQcdCcD8O2/qcSSSxjIdP6OdXMBdkiRJkka14qxl7Ln+Hl743VWs32MfVpzl1CBTzYSu4ydJkiRJ7VactYxDz1jMzC2PAjB30wZ2P2MxK4CFSxb1Njg9zh4/SZIk7VCD0jt0+c3ruPmeTfzb3Rt59dnXcPnN63odUlfsd+6Zjyd9Q2ZueZT9zj2zRxFpJGNdwP2do9WBs3pKkiRp4gald+jym9dx2qW38fOtjwGwbtNmTrv0NgCOO2JeL0ObdHtvun9C5eqNsXr83tHcfhv4a+Dk5vZp4P3dD02SJKnlxPNv4MTzb+h1GJoEg9I7dM7Vd7B5y9ZtyjZv2co5V9/Ro4i6Z8OsORMqV2+MmvhV1W81M3vOAF5UVe+qqncB85sySZIkaUIGpXfoR5s2T6h8Olu7+HQ2z9hlm7LNM3Zh7eLTexSRRtLJOX77VdV9bY9/DOzfpXgkSZLUxwald+g5s2ZOqHw6W7hkEauXnst9u+7FY4T1s/Zm9dJz+2robj/oJPH7WpKrk7wvyfuAfwK+Ot6TkixPsiHJ6lHqD0lyQ5JHkyweVveDJLclWZVkZScNkSRJ0tQ3KL1Dpx59MDNn7LRN2cwZO3Hq0Qf3KKLuWrhkERvn7s+3DzycuT/5sUnfFDTu5RyqalGS44HXNkUXVNVlHbz2hcAy4DOj1G8EPgQcN0r966vqgQ7+jyRJkqaJhUsWsQLY95MfY5+HN7Jh1hzWLj697xKFoQlcPvKFW/n51seYN2smpx59cN9N7KLpY8zEL8lOwJqqOgToJNl7XFVdl+SAMeo3ABuSvG0irytJkqTpbeGSRay5aDkb5+7P/DtvYm6vA+qS446Yx8XfugeAz3/wlT2ORoNuzKGeVbUVuCPJjj6nr4AvJ7kxySljLZjklCQrk6y8//7+OilYkiQNjkG5tp2k3hh3qCewB7AmybeAfx8qrKpf6VpU8JqqWpdkb+ArSb5TVdeNtGBVXQBcALBgwYLqYkySJEldMSjXtpPUO50kfjv8TNuqWtf83ZDkMuBlwIiJnyRJ0nQ35rXtTPwkTYJOJnf5+o4IZEiSZwJPq6qHm/tvAT6+I2OQJEnakQbl2naSemfcxC/JK4A/A14I/AKwE/DvVbXbOM+7GDgKmJ3kXmApzYXfq+q8JHOBlcBuwGNJPgy8CJgNXJZkKL6/q6qrnkzjJEmSpoMNs+Ywd9OGkct7EI+k/tPJUM9lwK8Bfw8sAN4DHDTek6rqpHHq1wP7jlD1EHBYB3FJkiT1hbWLT2f3tnP84Ilr25n4SZoMnVzAnaq6C9ipqrZW1d8Ax3Q3LEmSpMGxcMkiVi89l/t23YvHCOtn7c3qpec6sYukSdNJj98jSX4BWJXkT4D76DBhlCRJ3bPmoCMBmH/nTT2ORJNhUK5tJ6k3OkngfrNZbhGtyznsB7yrm0FJkiRJkiZPJ4nfa4EZVfVQVZ1RVf8dOKTLcUmSJEmSJkknid+fAd9I8sK2Mi+vIEmSJOlx8++8yaHnU1gnid/dwPuBLyR5d1OW7oUkSdJTs+agIx8//02SJHU2uUtV1U1JXgdcnOTltK7lJ0mSJD0p9gxJO1YnPX73AVTVA8DRQAGHdjMoSZIkSdLkGTfxq6q3td1/rKpOrSov5yBJkiRJ08S4Qz2TzAH+AHgR8PSh8qp6QxfjkiRJAuDym9dx8z2b+PnWx3j12ddw6tEHc9wR83odliRNK5303H0W+DbwPOAM4AfAii7GJEmSBLSSvtMuvY2fb30MgHWbNnPapbdx+c3rehyZJE0vnSR+e1XVXwNbqurrVfV+YNzeviTLk2xIsnqU+kOS3JDk0SSLh9Udk+SOJHcl+WhHLZEkaYCsOGsZe66/hxd+dxXr99iHFWct63VIXXHO1XewecvWbco2b9nKOVff0aOIJGl66mRWzy3N3/uSvA34EbBnB8+7EFgGfGaU+o3Ah4Dj2guT7AT8OfBm4F5gRZIrqur2Dv6nJGkUQ5c3cCa96W/FWcs49IzFzNzyKABzN21g9zMWswJYuGRRb4ObZD/atHlC5dOdn09J3dJJj9//SLI78PvAYuDTwP873pOq6jpayd1o9RuqagVPJJZDXgbcVVXfr6qfA58Dju0gTkmSBsJ+5575eNI3ZOaWR9nv3DN7FFH3PGfWzAmVS5JG1smsnl+qqp9W1eqqen1VvbSqruhiTPOAtW2P723KJEkSsPem+ydUPp2devTBzJyx7eWDZ87YiVOPPrhHEUnS9DTqUM8kf0brmn0jqqoPdSWiCUpyCnAKwP7779/jaCRJ6r4Ns+Ywd9OGkct7EE83Dc3e+ZEv3MrPtz7GvFkzndVTkp6EsXr8VgI3Nrdfabs/dOuWdcB+bY/3bcpGVFUXVNWCqlowZ86cLoYlSZoOBmHSk7WLT2fzjF22Kds8YxfWLj69RxF113FHzOOI/Wfx8uftyb9+9A0mfZL0JIza41dVFw3dT/Lh9sddtgI4MMnzaCV8vwb8+g7635KkaWxQJj1ZuGQRK4B9P/kx9nl4IxtmzWHt4tP7qo2SpMnVyayeMMaQz9EkuRg4Cpid5F5gKTADoKrOSzKXVq/ibsBjST4MvKiqHkqyCLga2AlYXlVrJvr/JUmDZ8xJT/osKVq4ZBFrLlrOxrn7M//Om/puiKckaXJ1mvhNWFWdNE79elrDOEequxK4shtxSZL61yBNeiJJ0kSMNbnLwzzR0/eMJA8NVQFVVbt1OzhJkiZikCY9kTQ9fP6Dr+x1CBIwxuQuVbVrVe3W3HZuu7+rSZ8kTS+DMOEJDN6kJ5IkdaqTC7hLkqaxoQlPnv3wgzyNYu6mDRx6xuK+TP4WLlnE6qXnct+ue/EYYf2svVm99FwnPZEkDTwTP0nqc2NOeNKHFi5ZxMa5+/PtAw9n7k9+bNInSRImfpLEieffwInn39DrMLrGCU8kSZKJnyT1uQ2z5kyoXJIk9R8TP0nqc054IkmSunYdP0nS1LBwySJWAPt+8mPs8/BGNsyaw9rFp3vuWx+Yf+dNvQ5BkjRNmPhJ0gBYuGQRay5azsa5+zP/zpu8pp0kSQPGoZ6SJEmS1OdM/CRJkiSpz5n4SZIkSVKf61ril2R5kg1JVo9SnySfSnJXkluTHNlWtzXJquZ2RbdilCRJkqRB0M0evwuBY8aofytwYHM7BfjLtrrNVXV4c/uV7oUoSZIkSf2va7N6VtV1SQ4YY5Fjgc9UVQHfTDIrybOr6r5uxSSpc2sOanXCO128piO3W0mSttXLc/zmAWvbHt/blAE8PcnKJN9MctxYL5LklGbZlffff3+XQpWecOL5N3Di+Tf0OgxJkiSpY1P1On7Prap1SZ4PXJPktqr63kgLVtUFwAUACxYsqB0ZpCRNJ/aCaTr7/Adf2esQJGla62WP3zpgv7bH+zZlVNXQ3+8D1wJH7OjgJEmSJKlf9DLxuwJ4TzO75yuAn1bVfUn2SLILQJLZwKuB23sYpzRwVpy1jD3X38MLv7uK9Xvsw4qzlvU6pK65/OZ13HzPJv7t7o28+uxruPzmdb0OSZIkadJ1bahnkouBo4DZSe4FlgIzAKrqPOBK4JeBu4BHgN9qnvpC4Pwkj9FKTM+uKhM/aQdZcdYyDj1jMTO3PArA3E0b2P2MxawAFi5Z1NvgJtnlN6/jtEtv4+dbHwNg3abNnHbpbQAcd8S8sZ4qSZI0raQ1qWZ/WLBgQa1cubLXYajPDU3s0q/nm6zfYx/mbtqwffmsvZn7kx/3IKLuefXZ17Bu0+btyufNmsm/fvQNPYhIkiTpqUlyY1UtGF7ey6GekqagvTeNPDvuaOXT2Y9GSPrGKpckSZquTPykCRiE88E2zJozofLp7DmzZk6oXJIkaboy8ZM6NNr5YP2W/K1dfDqbZ+yyTdnmGbuwdvHpPYqoe049+mBmzthpm7KZM3bi1KMP7lFEkiRJ3WHiJ3XonKvvYPOWrduUbd6ylXOuvqNHEXXHwiWLWL30XO7bdS8eI6yftTerl57bdxO7QGsCl0+888X8wk6tXeG8WTP5xDtf7MQukiSp70zVC7j3jX6fCGSQDNL5YAuXLGLNRcvZOHd/5t95E3N7HVAXHXfEPC7+1j2An1NJktS/7PHTpDjx/BseT3L7leeDSZIkaboy8ZM6NGjng82/8ybm33lTr8OQJEnSJDDx01M2CDNdgueDSZIkafryHD89JaPNdAn0ZULk+WCSJEmajuzx01MyKDNdSpIkSdOZiV8XDcIQyEGa6VKSJEmarrqa+CVZnmRDktWj1CfJp5LcleTWJEe21b03yXeb23u7GWc3DMrFvp3pUpIkSZr6ut3jdyFwzBj1bwUObG6nAH8JkGRPYCnwcuBlwNIke3Q10kk2KEMgB22mS0mSJGk66urkLlV1XZIDxljkWOAzVVXAN5PMSvJs4CjgK1W1ESDJV2glkBd3M97JNChDIIcmcPnIF27l51sfY96smZx69MF9ObHLECd1kSRJ0nTT61k95wFr2x7f25SNVj6mBx98kAsvvHCbsvnz57Nw4UK2bNnCZz/72e2ec/jhh3P44YfzyCOPcMkll2xXv2DBAg499FB++tOfctlll21X/8pXvpKDDz6YBx54gC996UuPl++285789D932m753Xbeuk2Mb3zjG9lvv/1Yu3YtX/va17Zb/phjjmHu3Ll8//vf57rrrtuu/u1vfzuzZ8/mjjvu4IYbtr+A+vHHH8/uu+/O6tWrWbly5Xb1J5xwAs94xjNYtWoVq1at2q7+5JNPZsaMGaxYsYI1a9ZsV/++972P446Yx96bf8idd94JwKZb7uHCW2DGjBmcfPLJAHz961/n7rvv3ua5z3jGMzjhhBMA+OpXv8q99967Tf1uu+3GO9/5TgCuuuoq1q9fv039XnvtxTve8Q4AvvjFL/Lggw9uUz937lyOOabV4XzppZfy0EMPbVO/77778qY3vQmASy65hEceeWSb+uc973m87nWvA+Czn/0sW7Zs2ab+oIMO4lWvehXAdtsd9G7bG/La176W5z//+axfv56rrrpqu/p+2PYArr/++se3vSET3fae++DtAFx4YatH3m3PbW9HbXvu97bltue2B257bntue+2e6rbXbtpP7pLklCQrk6wcvqJ66Y1z/p0ZqW3KZqR445x/71FEkiRJkgZVWqMsu/gPWkM9v1RVh45Qdz5wbVVd3Dy+g9Ywz6OAo6rqgyMtN5oFCxbUSL9y9MrlN68bqCGQkiRJknoryY1VtWB4ea97/K4A3tPM7vkK4KdVdR9wNfCWJHs0k7q8pSmbVo47Yh5H7D+Llz9vT/71o28w6ZMkSZLUE109xy/JxbR672YnuZfWTJ0zAKrqPOBK4JeBu4BHgN9q6jYmORNY0bzUx4cmepEkSZIkTUy3Z/U8aZz6An5nlLrlwPJuxCVJkiRJg6TXQz0lSZIkSV3W68s59D2v+SZJkiSp1+zxkyRJkqQ+Z+InSZIkSX3OxE+SJEmS+pyJnyRJkiT1ORM/SZIkSepzJn6SJEmS1OdM/CRJkiSpz6Wqeh3DpElyP/DDXscxgtnAA70OYgcYlHbC4LR1UNoJg9PWQWknDE5bB6WdMDhtHZR2wuC0dVDaCbZ1KnhuVc0ZXthXid9UlWRlVS3odRzdNijthMFp66C0EwanrYPSThictg5KO2Fw2joo7YTBaeugtBNs61TmUE9JkiRJ6nMmfpIkSZLU50z8dowLeh3ADjIo7YTBaeugtBMGp62D0k4YnLYOSjthcNo6KO2EwWnroLQTbOuU5Tl+kiRJktTn7PGTJEmSpD5n4jdBSZYkWZPk1iSrkrw8ybVJFjT1z0vy3SRHJ5mR5Ozm8U1Jbkjy1ma5HyT5xrDXXpVkdS/aNVySnw17/L4ky5r7H0vySJK9R1p+pPeoKR/1/ZgKhrXhl5PcmeS5TXsXN+UXJrm7adctSd7Y9pyXJbkuyR1Jbk7y6STP6EVbRjPCej1g+DbX3t7m8c5J7k9y9rDlrm3aemuS7yRZlmRWVxswARNta5JXJPm3Zt1+O8nH2pZ7a5KVSW5v1u2f7pBGdCDJ1qF9R5K/H9rmRltvbc9bleRzw8qGtu9bmu3/M0n23RHt6MQI+5Z/bv7eleSnzf1VSV7VLD8d2zg3yeeSfC/JjUmuTHJQkvlJrmk+c99NcnqSDHvutGvvkEnY/w7tj25J8q9JDu5FO0bT9jm9Ja3j39A2ekCSzU3d7c36mdHUjXnMTHJ4kkpyTK/aNZJx2jrid5wki5vjyKokK5K8p61udpItSf7rjmpDJ9raOXT7aFP+9uY4cUuzTj/Y9pz3NPvq25plxj3WTiUZdlxtyoZ/Rn91tOc0+7Ir27bpS5Ls0/3IO5Ntj6dfTPOdZtjndOjWvo2O+Flse701zfbw+0l6mnuZ+E1AklcCbweOrKqXAG8C1rbV7wtcBfx+VV0NnAk8Gzi0qo4EjgN2bXvJXZPs1zz3hTukEZPnAeD3hxeO8x6N935MCc2XiU8Bb62qka4LeWpVHQ58GDivec4+wN8Df1BVB1fVEbS2hSnXvifhzcCdwLuHf9EETm7W80uAR4F/3NHBTaKLgFOadXsocAlAkkOBZcBvVNWLgAXAXb0KcgSbq+rwqjoU+Dkw9OVo1PXW7G92An4pyTOHvd6pVXUYcDBwM3BNkl/oags6MMq+5eRmfX0A+EbzPhxeVddP0zYGuAy4tqpeUFUvBU4D9gGuAM6uqoOBw4BXAf9P23OnXXtH8mT2v21Obtp5EXBON+N8EoY+p4fRWqefaKv7XtOmFwP7Aic05eMdM08C/qX5O5WM1dbtNAndm4GXNe/DG4H2fda7gW8ydds5dDu7SdovAN7RtP8I4Fpo/YBIa7t9S1W9GHgF8NO21xvrWDvtJXk68E/AX1bVgc02/RfAdtea66H24+lG4Hfa6r43bH1/pq1utM/i0OvNp7V+3wos7WYDxmPiNzHPBh6oqkcBquqBqvpRW92XgSVVdUVav7r/F+B325b/cVVd0vZ6lwAnNvdPAi7eEY2YJMuBE5PsOax8xPeow/ej55K8Fvgr4O1V9b1xFr8BmNfc/x3goqq6Yaiyqr5QVT/uTqQ71EnA/wbuAV450gJV9XPgI8D+SQ7bgbFNpr2B+wCqamtV3d6UfwQ4q6q+01b3lz2KcTzfAH6xuT/WejsJ+Fta+6xjR3qhavmfwHpaB6teG2v/O5Lp2MbXA1uq6vGEpqpuAQ4C/rWqvtyUPQIsAj7a9tzp2N5tPIX973DX8cTnYCraDfjJ8MKq2gp8C5g33jGzSQzeDbwPeHPzpXoqGrGtw/wh8N+q6iGAqnqoqi5qqz+J1g/N8zIFe6uH2RXYGXgQoKoerao7mrrTgMVD+62m7q/anjvusXaa+3Xghqr64lBBVV1bVVNipNsIxtrHPK7Tz2JVbQBOARb1MrE38ZuYLwP7pTUE5S+SvK6t7iJgWVV9oXn8i8A9QzuyUfwD8M7m/juAL46x7I42s71LG/j4sPqf0Ur+fm9Y+WjvUSfvR6/tAlwOHDf0JX8cxzTLQ6uH6MbuhNU7zQ7sTbS2zYsZ4xfX5kvLLcAhOya6Sfc/gTuSXJbkg20772mxbpPsTOvL/G0drLcTgc+NUjfcTUyNdTrW/nck07GNo21r84eXN4nRs5Ls1hRNx/a2eyr73+HeAdw2OWFNmqFj6neAT9PqzdtG87l9Oa3RIuMdM18F3N1sB9cCb+tK1E/OuG0d0my/u1bV90ep3w94dlV9i21/LJ8KtvmelOTEqtpIq3f+h0kuTnJy29C+UY8lEznWTgPnDPv+OGRaHEsBkuxEq+f5irbiFwxb37/UlHf8WWy2851o/dDcEyZ+E1BVPwNeSitjvx/4fJL3NdVfBX4jEzun60HgJ0l+Dfg28MgkhvtUbTOEAfjjEZb5FPDeJI8PPRnnPZrqtgDXA789znLnJLkT+Dvgk12PqrtGm9Z3qPztwD9X1WZaP1Qc1+wQRzOVh6eM2daq+jitYZxfpvXL5FU7KK6namZzcF1J65fiv2aM9ZbW+cgPVNU9wNeAI0bouW83JdbpRPYt07WNT1aftHcy9r+fbT4LrwYWD39ijw0dUw+hlbR+pu1X/xc0cf8YuK+qbu3g9U6ilejT/J1KicJYbZ2oE2mG3TN12zl0+zxAVX2AVtLwLVrb4fIOXmuix9qp7NRh3x+nk6Hj6XpaQ+y/0lY3fKjn0DwdU/mzuB0TvwlqhnldW1VLaQ21eVdT9SfACuDvm1/e76I17G23UV5qyOeBP2d6DfMEoKo20Tr4/s6w8pHeo07fj156jNa5FS9L8odjLHdqVR0E/AFP7NDX0PpSOt08COwxrGxPWudwQmsH9qYkP6D1S91ewBtGeqHmIPViWj9iTEXjtZWq+l4zjPONwGFJ9mLqr9v2Lx+/2wy7HWu9nQQc0tR9j9ZQrHdt/7KPO4Ipsk7H2P8ON13bONq2dvvw8iTPB37W9AhN1/a2eyr73yEnN5+D46pq7QjPnRKaUwJm88S5TUPn+L0AeGmSX2GMY2azr30X8MfNOv8z4Jj2H2GnihHaOrz+IeBnzfY8kpOA9zXtvAJ4SZIDuxHrZKqq25ph1W/mic/iWMeSjo+109hUP5ZCczwFnkvrB7LfGWvhiX4Wm+18K7BhEmOeEBO/CUhy8LAdzuFA+8nnHwYeovWL++bm7/8eOok+yZwk7x72spfRShqv7lLY3fb/Ax+kNaZ91PeoOSelk/ejp5o43wacnGS8X56XAU9LcnRz/71pZjAFSPLOTKHZqkbS9KLcl+QNAE0vwTHAvzRfOH4J2L+qDqiqA2jtBLf7NSutE9o/Aazt8NfqHW6stjaP39b2q/SBtHbOm2hNEvGHSQ5qlntaptjscu3GWm/NkKMTgBe31R3LyOs0ST5E69y6nvd+drD/HVpu2rYRuAbYJckpQwVJXgLcAbwmyZuaspm0Rlz8yTRv7zaewv53WklyCK3hXg+2l1fVA7TO2zxtnGPmG4Fbq2q/Zp0/l1Yv0fE7sh2dGK2tw3wC+POhJDfJs9Ka/fIg4FlVNa9t2/4EU7hHpYn9qLaiw3liP/UJWj3Wc5tlfyHJByZyrJ3m/g54VZLHh0ImeW1aE6hNKc3n70PA7zedOaPp+LOYZA6tCamWVfXuIuomfhPzLOCitKbnvRV4EfCxocpmRb6X1kH1T4A/ojUk6fa0pjD+Eq3EkLbnPFxVn2x+pZ92mgPVZbTOz4Cx36Nx34+poBmjfwzwR80vr6MtV8D/AD5SrUlcfg04N60pxb8NHA08vCNinoBnJLm37fbfgfcApzfDG64BzmjGqh8PXFPNxAKNfwTekWRofX+2Wc+rgWcyyqQSPTKRtgL8Jq1z/FbRmiTj5KaH6VZaP+pc3KzX1cBov05PBaOuN1pfLtbVtpOiXAe8KMmzm8fnJLmF1uxyC4HXT5H905j73zbTto3NPuV4Wr/8fy/JGlpfFtfT+mz9UZI7aJ2/toJW8jNt2zuSJ7P/3VGxPUWPnw9Ga6TPe6t1XvRwl9Pad/0Sox8zT6J13G33D0ydRGGsth48bL/8buAvgX8GVjTt/AatHuBp087mdjatXqKPNN8DVgFn0Jr0g6q6ktZn9qvNZ/smWr3znRxrp4qRjqsdaYaxvh343bQu53A7rZmJ7+9WsE9FVd0M3MoT29vwc/w+xPjb6NA2sobWKWFfprVN9Ex6mHRKkiRJknYAe/wkSZIkqc+Z+EmSJElSnzPxkyRJkqQ+Z+InSZIkSX3OxE+SJEmS+pyJnySpLyWZm+RzzaURbkxy5dD1GHsY02XN9N53Jflp29Tgr+plXJKk/uflHCRJfSdJgOuBi6rqvKbsMGC3qvrGDo4jVfXYsPKjgMVV9fYdFYskabDZ4ydJ6kevB7YMJX0AVXVLVX0jybOSfC3JTUluS3IsQJIDknw7yV8lWZPky0lmNnW/mOSrSW5pnveCpvzUJCuS3JrkjLbXuSPJZ4DVwH5jBZrkuiSHtz3+lySHJflYkr9NckNzweP/0rbMdv9XkqSxmPhJkvrRocCNo9T9B3B8VR1JK0H806ZnDuBA4M+raj6wCXhXU/7Zpvww4FXAfUne0iz/MuBw4KVJXtv2On9RVfOr6ofjxPrXwPsAmqGoT6+qW5q6lwBvAF4J/HGS54zzfyVJGpGJnyRp0AT4/5LcCnwVmAfs09TdXVWrmvs3Agck2RWYV1WXAVTVf1TVI8BbmtvNwE3AIbQSMoAfVtU3O4zn74G3J5kBvB+4sK3uH6tqc1U9APwzrWRvrP8rSdKIdu51AJIkdcEa4FdHqTsZmAO8tKq2JPkB8PSm7tG25bYCM8f4HwE+UVXnb1OYHAD8e6eBVtUjSb4CHAucALy0vXr44qP9X0mSxmKPnySpH10D7JLklKGCJC9J8kvA7sCGJul7PfDcsV6oqh4G7k1yXPM6uyR5BnA18P4kz2rK5yXZ+0nG+2ngU8CKqvpJW/mxSZ6eZC/gKGDFJP9fSdKAsMdPktR3qqqSHA/8ryR/QOu8vh8AH6Z1vt4Xk9wGrAS+08FL/iZwfpKPA1uAd1fVl5O8ELihOUXwZ8Bv0OopnGi8NyZ5CPibYVW30hriORs4s6p+BPxolP+7YaL/V5I0OLycgyRJPZbkOcC1wCFDl35I8jHgZ1V1bg9DkyT1CYd6SpLUQ0neA/wbsGT49f4kSZos9vhJkiRJUp+zx0+SJEmS+pyJnyRJkiT1ORM/SZIkSepzJn6SJEmS1OdM/CRJkiSpz5n4SZIkSVKf+7+0LoVPoHIsOQAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<Figure size 1080x216 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "haz_plot_=hazard_list\n",
    "haz_plot_.index=cancer_list\n",
    "haz_plot_[\"Cancer Type\"]=cancer_list\n",
    "#haz_plot_=haz_plot_.drop(haz_plot_.index[abs(haz_plot_[\"coef\"])>1][0])\n",
    "haz_plot_=haz_plot_.drop([\"KICH\"])\n",
    "haz_plot_=fdr(haz_plot_,False,\"p\")\n",
    "haz_plot_=haz_plot_.sort_values(by=\"exp(coef)\",ascending=True)\n",
    "plt.figure(figsize=(15,3))\n",
    "plt.errorbar(x=haz_plot_[\"Cancer Type\"].tolist(),y=haz_plot_[\"exp(coef)\"].tolist(),yerr=haz_plot_[\"se(coef)\"].tolist(),fmt=\"o\")\n",
    "haz_plot_thresh=haz_plot_[haz_plot_[\"fdr\"]<0.05]\n",
    "plt.errorbar(x=haz_plot_thresh[\"Cancer Type\"].tolist(),y=haz_plot_thresh[\"exp(coef)\"].tolist(),yerr=haz_plot_thresh[\"se(coef)\"].tolist(),fmt=\"o\",color=\"red\")\n",
    "plt.xlabel(\"Cancer Type\")\n",
    "plt.ylabel(\"Hazard Ratio\")\n",
    "plt.axhline(1,linestyle=\"--\",color=\"gray\")\n",
    "#plt.savefig(\"/cellar/users/tsears/projects/germline-immune/plots/fig5/cancer_risk_PFI_hazardV2_long.pdf\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "449f8af9-a457-4b39-af1c-a2ddc38df5b6",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "ML_kernal",
   "language": "python",
   "name": "ml_kernal"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
